<!DOCTYPE html>
<html lang="ko" xml:lang="ko">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>3 GWAS ì ˆì°¨ | GWAS ì ˆì°¨</title>
  <meta name="description" content="3 GWAS ì ˆì°¨ | GWAS ì ˆì°¨" />
  <meta name="generator" content="bookdown 0.44 and GitBook 2.6.7" />

  <meta property="og:title" content="3 GWAS ì ˆì°¨ | GWAS ì ˆì°¨" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="3 GWAS ì ˆì°¨ | GWAS ì ˆì°¨" />
  
  
  



<meta name="date" content="2025-09-08" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="pre-gwas-ì ˆì°¨-ìœ ì „ìí˜•-ë°ì´í„°-qc.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path=""><a href="#%EC%A0%80%EC%9E%90-%EC%A0%95%EB%B3%B4"><i class="fa fa-check"></i>ì €ì ì •ë³´</a></li>
<li class="chapter" data-level="1" data-path="02-gwas.html"><a href="#gwas-%EB%B6%84%EC%84%9D-%EC%82%AC%EC%A0%84-%EC%A4%80%EB%B9%84-%EC%A0%88%EC%B0%A8"><i class="fa fa-check"></i><b>1</b> GWAS ë¶„ì„ ì‚¬ì „ ì¤€ë¹„ ì ˆì°¨</a>
<ul>
<li class="chapter" data-level="1.1" data-path="02-gwas.html"><a href="#genotype-data-%EC%9C%A0%EC%A0%84%EC%9E%90%ED%98%95-%EB%8D%B0%EC%9D%B4%ED%84%B0-%EC%A4%80%EB%B9%84"><i class="fa fa-check"></i><b>1.1</b> Genotype data (ìœ ì „ìí˜• ë°ì´í„°) ì¤€ë¹„</a></li>
<li class="chapter" data-level="1.2" data-path="02-gwas.html"><a href="#phenotype-data-%ED%91%9C%ED%98%84%ED%98%95-%EB%8D%B0%EC%9D%B4%ED%84%B0-%EC%A4%80%EB%B9%84"><i class="fa fa-check"></i><b>1.2</b> Phenotype data (í‘œí˜„í˜• ë°ì´í„°) ì¤€ë¹„</a></li>
<li class="chapter" data-level="1.3" data-path="02-gwas.html"><a href="#rrstudio-%ED%94%84%EB%A1%9C%EA%B7%B8%EB%9E%A8-%EB%8B%A4%EC%9A%B4%EB%A1%9C%EB%93%9C-%EB%B0%8F-%EC%84%A4%EC%B9%98"><i class="fa fa-check"></i><b>1.3</b> R/Rstudio í”„ë¡œê·¸ë¨ ë‹¤ìš´ë¡œë“œ ë° ì„¤ì¹˜</a></li>
<li class="chapter" data-level="1.4" data-path="02-gwas.html"><a href="#plink2-%ED%94%84%EB%A1%9C%EA%B7%B8%EB%9E%A8-%EB%8B%A4%EC%9A%B4%EB%A1%9C%EB%93%9C-%EB%B0%8F-%EC%84%A4%EC%B9%98"><i class="fa fa-check"></i><b>1.4</b> plink2 í”„ë¡œê·¸ë¨ ë‹¤ìš´ë¡œë“œ ë° ì„¤ì¹˜</a></li>
<li class="chapter" data-level="1.5" data-path="02-gwas.html"><a href="#r-package-%EC%84%A4%EC%B9%98"><i class="fa fa-check"></i><b>1.5</b> R package ì„¤ì¹˜</a></li>
<li class="chapter" data-level="1.6" data-path="02-gwas.html"><a href="#gwas-%EB%B6%84%EC%84%9D-%EC%82%AC%EC%A0%84-%EC%A4%80%EB%B9%84-%EC%99%84%EB%A3%8C"><i class="fa fa-check"></i><b>1.6</b> ğŸ‰ GWAS ë¶„ì„ ì‚¬ì „ ì¤€ë¹„ ì™„ë£Œ</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="02-gwas.html"><a href="#pre-gwas-%EC%A0%88%EC%B0%A8-%EC%9C%A0%EC%A0%84%EC%9E%90%ED%98%95-%EB%8D%B0%EC%9D%B4%ED%84%B0-qc"><i class="fa fa-check"></i><b>2</b> Pre-GWAS ì ˆì°¨: ìœ ì „ìí˜• ë°ì´í„° QC</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-gwas.html"><a href="#rstudio-terminal-%EC%A0%84%EC%B2%B4-%EC%9C%A0%EC%A0%84%EC%9E%90%ED%98%95-%EB%8D%B0%EC%9D%B4%ED%84%B0-qc-%EC%9B%8C%ED%81%AC%ED%94%8C%EB%A1%9C%EC%9A%B0"><i class="fa fa-check"></i><b>2.1</b> [RStudio Terminal] ì „ì²´ ìœ ì „ìí˜• ë°ì´í„° QC ì›Œí¬í”Œë¡œìš°</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="02-gwas.html"><a href="#%EA%B2%B0%EA%B3%BC-%EC%A0%80%EC%9E%A5-%EB%94%94%EB%A0%89%ED%84%B0%EB%A6%AC-%EC%A4%80%EB%B9%84"><i class="fa fa-check"></i><b>2.1.1</b> 0) ê²°ê³¼ ì €ì¥ ë””ë ‰í„°ë¦¬ ì¤€ë¹„</a></li>
<li class="chapter" data-level="2.1.2" data-path="02-gwas.html"><a href="#%EC%9B%90%EB%B3%B8-pedmap%EC%9D%84-plink2-pgen-%EC%84%B8%ED%8A%B8%EB%A1%9C-%EB%B3%80%ED%99%98"><i class="fa fa-check"></i><b>2.1.2</b> 1) ì›ë³¸ PED/MAPì„ PLINK2 PGEN ì„¸íŠ¸ë¡œ ë³€í™˜</a></li>
<li class="chapter" data-level="2.1.3" data-path="02-gwas.html"><a href="#refalt-.-%EC%98%A4%EB%A5%98-%EB%B3%80%EC%9D%B4-%EC%A0%9C%EA%B1%B0"><i class="fa fa-check"></i><b>2.1.3</b> 2) REF/ALT <code>.</code> ì˜¤ë¥˜ ë³€ì´ ì œê±°</a></li>
<li class="chapter" data-level="2.1.4" data-path="02-gwas.html"><a href="#%EC%B0%A8-%ED%95%84%ED%84%B0-%EC%A0%81%EC%9A%A9"><i class="fa fa-check"></i><b>2.1.4</b> 3) 1ì°¨ í•„í„° ì ìš©</a></li>
<li class="chapter" data-level="2.1.5" data-path="02-gwas.html"><a href="#%ED%92%88%EC%A2%85%EB%B3%84-iid-%EB%AA%A9%EB%A1%9D-%EC%A4%80%EB%B9%84"><i class="fa fa-check"></i><b>2.1.5</b> 4) í’ˆì¢…ë³„ IID ëª©ë¡ ì¤€ë¹„</a></li>
<li class="chapter" data-level="2.1.6" data-path="02-gwas.html"><a href="#%ED%92%88%EC%A2%85%EB%B3%84-%EC%84%9C%EB%B8%8C%EC%85%8B-pfile-%EC%83%9D%EC%84%B1"><i class="fa fa-check"></i><b>2.1.6</b> 5) í’ˆì¢…ë³„ ì„œë¸Œì…‹ pfile ìƒì„±</a></li>
<li class="chapter" data-level="2.1.7" data-path="02-gwas.html"><a href="#%EC%B0%A8-%ED%95%84%ED%84%B0-%EC%A0%81%EC%9A%A9-%EB%B0%8F-%EC%B5%9C%EC%A2%85-%EB%8D%B0%EC%9D%B4%ED%84%B0%EC%85%8B-%EC%83%9D%EC%84%B1"><i class="fa fa-check"></i><b>2.1.7</b> 6) 2ì°¨ í•„í„° ì ìš© ë° ìµœì¢… ë°ì´í„°ì…‹ ìƒì„±</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="02-gwas.html"><a href="#%ED%92%88%EC%A2%85%EB%B3%84-qc-%EB%A6%AC%ED%8F%AC%ED%8A%B8-%ED%99%9C%EC%9A%A9-2%EC%B0%A8-%ED%95%84%ED%84%B0-%EA%B8%B0%EC%A4%80%EA%B0%92-%EC%A0%95%ED%95%98%EA%B8%B0"><i class="fa fa-check"></i><b>2.2</b> í’ˆì¢…ë³„ QC ë¦¬í¬íŠ¸ í™œìš© 2ì°¨ í•„í„° ê¸°ì¤€ê°’ ì •í•˜ê¸°</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="02-gwas.html"><a href="#rstudio-terminal-%ED%92%88%EC%A2%85%EB%B3%84-qc-%EB%A6%AC%ED%8F%AC%ED%8A%B8-%EC%B6%9C%EB%A0%A5"><i class="fa fa-check"></i><b>2.2.1</b> 1) [RStudio Terminal] í’ˆì¢…ë³„ QC ë¦¬í¬íŠ¸ ì¶œë ¥</a></li>
<li class="chapter" data-level="2.2.2" data-path="02-gwas.html"><a href="#rstudio-console-%EC%84%A4%EC%A0%95"><i class="fa fa-check"></i><b>2.2.2</b> 2) [RStudio Console] ì„¤ì •</a></li>
<li class="chapter" data-level="2.2.3" data-path="02-gwas.html"><a href="#rstudio-console-%ED%92%88%EC%A2%85%EB%B3%84-qc-%EB%A6%AC%ED%8F%AC%ED%8A%B8-%EC%A0%90%EA%B2%80"><i class="fa fa-check"></i><b>2.2.3</b> 3) [RStudio Console] í’ˆì¢…ë³„ QC ë¦¬í¬íŠ¸ ì ê²€</a></li>
<li class="chapter" data-level="2.2.4" data-path="02-gwas.html"><a href="#rstudio-console-%ED%92%88%EC%A2%85%EB%B3%84-qc-%EB%A6%AC%ED%8F%AC%ED%8A%B8-%EC%9A%94%EC%95%BD-%EA%B2%B0%EA%B3%BC-%ED%86%B5%ED%95%A9"><i class="fa fa-check"></i><b>2.2.4</b> 4) [RStudio Console] í’ˆì¢…ë³„ QC ë¦¬í¬íŠ¸ ìš”ì•½ ê²°ê³¼ í†µí•©</a></li>
<li class="chapter" data-level="2.2.5" data-path="02-gwas.html"><a href="#qc-%EB%A6%AC%ED%8F%AC%ED%8A%B8-%EA%B2%B0%EA%B3%BC-%EB%8B%A4%EC%8B%9C%EB%B3%B4%EA%B8%B0"><i class="fa fa-check"></i><b>2.2.5</b> 5) QC ë¦¬í¬íŠ¸ ê²°ê³¼ ë‹¤ì‹œë³´ê¸°</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="02-gwas.html"><a href="#%EC%B5%9C%EC%A2%85-plink1-binary-%EC%84%B8%ED%8A%B8-%EB%8B%A4%EC%8B%9C-%EB%A7%8C%EB%93%A4%EA%B8%B0"><i class="fa fa-check"></i><b>2.3</b> ìµœì¢… PLINK1 Binary ì„¸íŠ¸ ë‹¤ì‹œ ë§Œë“¤ê¸°</a></li>
<li class="chapter" data-level="2.4" data-path="02-gwas.html"><a href="#qc-%EA%B3%BC%EC%A0%95-%EC%99%84%EB%A3%8C"><i class="fa fa-check"></i><b>2.4</b> ğŸ‰ QC ê³¼ì • ì™„ë£Œ</a></li>
<li class="chapter" data-level="2.5" data-path="02-gwas.html"><a href="#%EC%8A%A4%ED%81%AC%EB%A6%BD%ED%8A%B8-%EC%A0%84%EC%B2%B4-%EB%B3%B4%EA%B8%B0"><i class="fa fa-check"></i><b>2.5</b> ìŠ¤í¬ë¦½íŠ¸ ì „ì²´ ë³´ê¸°</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="02-gwas.html"><a href="#gwas-%EC%A0%88%EC%B0%A8"><i class="fa fa-check"></i><b>3</b> GWAS ì ˆì°¨</a>
<ul>
<li class="chapter" data-level="3.1" data-path="02-gwas.html"><a href="#%ED%8C%A8%ED%82%A4%EC%A7%80-%EC%84%A4%EC%B9%98-%EB%A1%9C%EB%94%A9"><i class="fa fa-check"></i><b>3.1</b> íŒ¨í‚¤ì§€ ì„¤ì¹˜ &amp; ë¡œë”©</a></li>
<li class="chapter" data-level="3.2" data-path="02-gwas.html"><a href="#%EA%B2%BD%EB%A1%9C%EC%9E%85%EB%A0%A5-%EC%A0%95%EC%9D%98"><i class="fa fa-check"></i><b>3.2</b> ê²½ë¡œ/ì…ë ¥ ì •ì˜</a></li>
<li class="chapter" data-level="3.3" data-path="02-gwas.html"><a href="#plink-binary-%EB%8D%B0%EC%9D%B4%ED%84%B0-%EB%B6%88%EB%9F%AC%EC%98%A4%EA%B8%B0"><i class="fa fa-check"></i><b>3.3</b> PLINK binary ë°ì´í„° ë¶ˆëŸ¬ì˜¤ê¸°</a></li>
<li class="chapter" data-level="3.4" data-path="02-gwas.html"><a href="#%ED%91%9C%ED%98%84%ED%98%95-%EB%8D%B0%EC%9D%B4%ED%84%B0-%EB%A1%9C%EB%94%A9-%EC%A0%95%EB%A0%AC"><i class="fa fa-check"></i><b>3.4</b> í‘œí˜„í˜• ë°ì´í„° ë¡œë”© &amp; ì •ë ¬</a></li>
<li class="chapter" data-level="3.5" data-path="02-gwas.html"><a href="#grm%EC%9C%A0%EC%A0%84%EC%B2%B4-%EA%B4%80%EA%B3%84-%ED%96%89%EB%A0%AC-%EA%B3%84%EC%82%B0-%EA%B0%84%EB%8B%A8-%EC%8B%9C%EA%B0%81%ED%99%94"><i class="fa fa-check"></i><b>3.5</b> GRM(ìœ ì „ì²´ ê´€ê³„ í–‰ë ¬) ê³„ì‚° &amp; ê°„ë‹¨ ì‹œê°í™”</a></li>
<li class="chapter" data-level="3.6" data-path="02-gwas.html"><a href="#gwas-%EB%B6%84%EC%84%9D%EC%9A%A9-%EB%8D%B0%EC%9D%B4%ED%84%B0-%EA%B0%9D%EC%B2%B4-%EB%A7%8C%EB%93%A4%EA%B8%B0"><i class="fa fa-check"></i><b>3.6</b> GWAS ë¶„ì„ìš© ë°ì´í„° ê°ì²´ ë§Œë“¤ê¸°</a></li>
<li class="chapter" data-level="3.7" data-path="02-gwas.html"><a href="#gwas-%EA%B7%80%EB%AC%B4%EB%AA%A8%ED%98%95-%EC%A0%81%ED%95%A9-%EB%B0%8F-%EC%9C%A0%EC%A0%84%EB%A0%A5snp-heritability-%EA%B3%84%EC%82%B0"><i class="fa fa-check"></i><b>3.7</b> GWAS ê·€ë¬´ëª¨í˜• ì í•© ë° ìœ ì „ë ¥(SNP heritability) ê³„ì‚°</a>
<ul>
<li class="chapter" data-level="3.7.1" data-path="02-gwas.html"><a href="#%ED%95%A8%EC%88%98%EB%A5%BC-%EC%9D%B4%EC%9A%A9%ED%95%9C-%EC%9C%A0%EC%A0%84%EB%A0%A5-%EA%B3%84%EC%82%B0"><i class="fa fa-check"></i><b>3.7.1</b> í•¨ìˆ˜ë¥¼ ì´ìš©í•œ ìœ ì „ë ¥ ê³„ì‚°</a></li>
</ul></li>
<li class="chapter" data-level="3.8" data-path="02-gwas.html"><a href="#gwas-%EC%B5%9C%EC%A2%85-null-%EB%AA%A8%ED%98%95-%EC%A0%81%ED%95%A9-%EB%B0%8F-gwas"><i class="fa fa-check"></i><b>3.8</b> GWAS ìµœì¢… null ëª¨í˜• ì í•© ë° GWAS</a>
<ul>
<li class="chapter" data-level="3.8.1" data-path="gwas-ì ˆì°¨.html"><a href="gwas-ì ˆì°¨.html"><i class="fa fa-check"></i><b>3.8.1</b> Null modelì´ë€? ì™œ ë¨¼ì € ì í•©í•˜ëŠ”ê°€?</a></li>
</ul></li>
<li class="chapter" data-level="3.9" data-path="02-gwas.html"><a href="#gwas-%EA%B2%B0%EA%B3%BC-%EC%8B%9C%EA%B0%81%ED%99%94"><i class="fa fa-check"></i><b>3.9</b> GWAS ê²°ê³¼ ì‹œê°í™”</a></li>
<li class="chapter" data-level="3.10" data-path="02-gwas.html"><a href="#r%EB%A1%9C-gwas-%EA%B3%BC%EC%A0%95-%EC%99%84%EB%A3%8C"><i class="fa fa-check"></i><b>3.10</b> ğŸ‰ Rë¡œ GWAS ê³¼ì • ì™„ë£Œ</a></li>
<li class="chapter" data-level="3.11" data-path="gwas-ì ˆì°¨.html"><a href="gwas-ì ˆì°¨.html#script-full"><i class="fa fa-check"></i><b>3.11</b> ìŠ¤í¬ë¦½íŠ¸ ì „ì²´ ë³´ê¸°</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">GWAS ì ˆì°¨</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="gwas-ì ˆì°¨" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">3</span> GWAS ì ˆì°¨<a href="#gwas-%EC%A0%88%EC%B0%A8" class="anchor-section" aria-label="Anchor link to header"></a></h1>
ê°œìš”
<div class="alert alert-info checklist" role="alert">
<p>
<span class="icon">ğŸ“‹</span> <strong>ë‹¨ê³„ë³„ íŠœí† ë¦¬ì–¼: Rì—ì„œ GWAS ìˆ˜í–‰</strong>
</p>
<ul class="tight">
<li>
í•„ìš” íŒ¨í‚¤ì§€ ì„¤ì¹˜/ë¡œë”©
</li>
<li>
PLINK ìœ ì „ìí˜• ë°ì´í„° &amp; í‘œí˜„í˜• ë°ì´í„° ë¡œë”© ë° ì •ë¦¬
</li>
<li>
GRM ê³„ì‚° ë° ì‹œê°í™”
</li>
<li>
lme4breeding + GWASTools + GENESIS ê¸°ë°˜ GWAS ê·€ë¬´ëª¨í˜•(Null model) ì í•©
</li>
<li>
ë‹¨ì¼ ë§ˆì»¤ ì—°ê´€ì„± ê²€ì •, P ê°’ ë³´ì •, ê²°ê³¼ ì‹œê°í™”
</li>
</ul>
</div>
<div id="íŒ¨í‚¤ì§€-ì„¤ì¹˜-ë¡œë”©" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> íŒ¨í‚¤ì§€ ì„¤ì¹˜ &amp; ë¡œë”©<a href="#%ED%8C%A8%ED%82%A4%EC%A7%80-%EC%84%A4%EC%B9%98-%EB%A1%9C%EB%94%A9" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="gwas-ì ˆì°¨.html#cb26-1" tabindex="-1"></a><span class="co"># ---- ì„¤ì¹˜ì–´ì‹œìŠ¤í„´íŠ¸ -------------------------------------------------------</span></span>
<span id="cb26-2"><a href="gwas-ì ˆì°¨.html#cb26-2" tabindex="-1"></a>cran_pkgs <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;genio&quot;</span>, <span class="st">&quot;qqman&quot;</span>, <span class="st">&quot;sommer&quot;</span>, <span class="st">&quot;lme4breeding&quot;</span>, <span class="st">&quot;dplyr&quot;</span>, <span class="st">&quot;CMplot&quot;</span>, <span class="st">&quot;stringr&quot;</span>, <span class="st">&quot;ggplot2&quot;</span>, <span class="st">&quot;tibble&quot;</span>, <span class="st">&quot;readr&quot;</span>)</span>
<span id="cb26-3"><a href="gwas-ì ˆì°¨.html#cb26-3" tabindex="-1"></a>bioc_pkgs <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;GENESIS&quot;</span>,<span class="st">&quot;GWASTools&quot;</span>,<span class="st">&quot;BiocParallel&quot;</span>)</span>
<span id="cb26-4"><a href="gwas-ì ˆì°¨.html#cb26-4" tabindex="-1"></a></span>
<span id="cb26-5"><a href="gwas-ì ˆì°¨.html#cb26-5" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) <span class="fu">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb26-6"><a href="gwas-ì ˆì°¨.html#cb26-6" tabindex="-1"></a></span>
<span id="cb26-7"><a href="gwas-ì ˆì°¨.html#cb26-7" tabindex="-1"></a>install_if_missing <span class="ot">&lt;-</span> <span class="cf">function</span>(pkgs, installer) {</span>
<span id="cb26-8"><a href="gwas-ì ˆì°¨.html#cb26-8" tabindex="-1"></a>  need <span class="ot">&lt;-</span> pkgs[<span class="sc">!</span><span class="fu">vapply</span>(pkgs, requireNamespace, <span class="fu">logical</span>(<span class="dv">1</span>), <span class="at">quietly =</span> <span class="cn">TRUE</span>)]</span>
<span id="cb26-9"><a href="gwas-ì ˆì°¨.html#cb26-9" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">length</span>(need)) <span class="fu">installer</span>(need, <span class="at">ask =</span> <span class="cn">FALSE</span>)</span>
<span id="cb26-10"><a href="gwas-ì ˆì°¨.html#cb26-10" tabindex="-1"></a>}</span>
<span id="cb26-11"><a href="gwas-ì ˆì°¨.html#cb26-11" tabindex="-1"></a></span>
<span id="cb26-12"><a href="gwas-ì ˆì°¨.html#cb26-12" tabindex="-1"></a><span class="fu">install_if_missing</span>(cran_pkgs, install.packages)</span>
<span id="cb26-13"><a href="gwas-ì ˆì°¨.html#cb26-13" tabindex="-1"></a><span class="fu">install_if_missing</span>(bioc_pkgs, BiocManager<span class="sc">::</span>install)</span>
<span id="cb26-14"><a href="gwas-ì ˆì°¨.html#cb26-14" tabindex="-1"></a></span>
<span id="cb26-15"><a href="gwas-ì ˆì°¨.html#cb26-15" tabindex="-1"></a><span class="co"># ---- ë¼ì´ë¸ŒëŸ¬ë¦¬ ë¡œë“œ ------------------------------------------------------</span></span>
<span id="cb26-16"><a href="gwas-ì ˆì°¨.html#cb26-16" tabindex="-1"></a><span class="fu">library</span>(genio)</span>
<span id="cb26-17"><a href="gwas-ì ˆì°¨.html#cb26-17" tabindex="-1"></a><span class="fu">library</span>(GENESIS)</span>
<span id="cb26-18"><a href="gwas-ì ˆì°¨.html#cb26-18" tabindex="-1"></a><span class="fu">library</span>(GWASTools)</span>
<span id="cb26-19"><a href="gwas-ì ˆì°¨.html#cb26-19" tabindex="-1"></a><span class="fu">library</span>(qqman)</span>
<span id="cb26-20"><a href="gwas-ì ˆì°¨.html#cb26-20" tabindex="-1"></a><span class="fu">library</span>(CMplot)</span>
<span id="cb26-21"><a href="gwas-ì ˆì°¨.html#cb26-21" tabindex="-1"></a><span class="fu">library</span>(lme4breeding)</span>
<span id="cb26-22"><a href="gwas-ì ˆì°¨.html#cb26-22" tabindex="-1"></a><span class="fu">library</span>(sommer)</span>
<span id="cb26-23"><a href="gwas-ì ˆì°¨.html#cb26-23" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb26-24"><a href="gwas-ì ˆì°¨.html#cb26-24" tabindex="-1"></a><span class="fu">library</span>(stringr)</span>
<span id="cb26-25"><a href="gwas-ì ˆì°¨.html#cb26-25" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb26-26"><a href="gwas-ì ˆì°¨.html#cb26-26" tabindex="-1"></a><span class="fu">library</span>(tibble)</span></code></pre></div>
</div>
<div id="ê²½ë¡œì…ë ¥-ì •ì˜" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> ê²½ë¡œ/ì…ë ¥ ì •ì˜<a href="#%EA%B2%BD%EB%A1%9C%EC%9E%85%EB%A0%A5-%EC%A0%95%EC%9D%98" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="gwas-ì ˆì°¨.html#cb27-1" tabindex="-1"></a>breed        <span class="ot">&lt;-</span> <span class="st">&quot;holstein&quot;</span></span>
<span id="cb27-2"><a href="gwas-ì ˆì°¨.html#cb27-2" tabindex="-1"></a>plink_prefix <span class="ot">&lt;-</span> <span class="st">&quot;03_qc/NIAS_ibv3_296ea.holstein&quot;</span></span>
<span id="cb27-3"><a href="gwas-ì ˆì°¨.html#cb27-3" tabindex="-1"></a>pheno_file   <span class="ot">&lt;-</span> <span class="st">&quot;NIAS_ibv3_296ea_pheno.csv&quot;</span></span>
<span id="cb27-4"><a href="gwas-ì ˆì°¨.html#cb27-4" tabindex="-1"></a></span>
<span id="cb27-5"><a href="gwas-ì ˆì°¨.html#cb27-5" tabindex="-1"></a>report_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">&quot;05_gwas&quot;</span>, breed)</span>
<span id="cb27-6"><a href="gwas-ì ˆì°¨.html#cb27-6" tabindex="-1"></a>figs_dir   <span class="ot">&lt;-</span> <span class="fu">file.path</span>(report_dir, <span class="st">&quot;figs&quot;</span>)</span>
<span id="cb27-7"><a href="gwas-ì ˆì°¨.html#cb27-7" tabindex="-1"></a>gwas_dir   <span class="ot">&lt;-</span> <span class="fu">file.path</span>(report_dir, <span class="st">&quot;gwas&quot;</span>)</span>
<span id="cb27-8"><a href="gwas-ì ˆì°¨.html#cb27-8" tabindex="-1"></a>tables_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(report_dir, <span class="st">&quot;tables&quot;</span>)</span>
<span id="cb27-9"><a href="gwas-ì ˆì°¨.html#cb27-9" tabindex="-1"></a><span class="cf">for</span> (d <span class="cf">in</span> <span class="fu">c</span>(report_dir, figs_dir, gwas_dir, tables_dir)) <span class="fu">dir.create</span>(d, <span class="at">recursive =</span> <span class="cn">TRUE</span>, <span class="at">showWarnings =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</div>
<div id="plink-binary-ë°ì´í„°-ë¶ˆëŸ¬ì˜¤ê¸°" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> PLINK binary ë°ì´í„° ë¶ˆëŸ¬ì˜¤ê¸°<a href="#plink-binary-%EB%8D%B0%EC%9D%B4%ED%84%B0-%EB%B6%88%EB%9F%AC%EC%98%A4%EA%B8%B0" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="gwas-ì ˆì°¨.html#cb28-1" tabindex="-1"></a><span class="co"># read_plink()ëŠ” SNP x Sample í–‰ë ¬(X), bim(ë§ˆì»¤ì •ë³´), fam(ê°œì²´ì •ë³´)ì„ ë¶ˆëŸ¬ì˜¨ë‹¤.</span></span>
<span id="cb28-2"><a href="gwas-ì ˆì°¨.html#cb28-2" tabindex="-1"></a>plink <span class="ot">&lt;-</span> <span class="fu">read_plink</span>(plink_prefix)</span>
<span id="cb28-3"><a href="gwas-ì ˆì°¨.html#cb28-3" tabindex="-1"></a></span>
<span id="cb28-4"><a href="gwas-ì ˆì°¨.html#cb28-4" tabindex="-1"></a>X <span class="ot">&lt;-</span> plink<span class="sc">$</span>X             <span class="co"># í–‰: SNP, ì—´: ìƒ˜í”Œ, ê°’: 0/1/2</span></span>
<span id="cb28-5"><a href="gwas-ì ˆì°¨.html#cb28-5" tabindex="-1"></a>bim <span class="ot">&lt;-</span> plink<span class="sc">$</span>bim         <span class="co"># chr, id, pos, alt, ref ë“±</span></span>
<span id="cb28-6"><a href="gwas-ì ˆì°¨.html#cb28-6" tabindex="-1"></a>fam <span class="ot">&lt;-</span> plink<span class="sc">$</span>fam         <span class="co"># family/sample id ë“±</span></span>
<span id="cb28-7"><a href="gwas-ì ˆì°¨.html#cb28-7" tabindex="-1"></a></span>
<span id="cb28-8"><a href="gwas-ì ˆì°¨.html#cb28-8" tabindex="-1"></a><span class="co"># ë¯¸ë¦¬ë³´ê¸°</span></span>
<span id="cb28-9"><a href="gwas-ì ˆì°¨.html#cb28-9" tabindex="-1"></a>X[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>#&gt;                                 3530 3528 3527 3195 3994
#&gt; Hapmap43437-BTA-101873             1    1    1    1    0
#&gt; ARS-BFGL-NGS-16466                 0    0    0    0    0
#&gt; ARS-BFGL-NGS-105096                1    0    1    0    0
#&gt; Hapmap34944-BES1_Contig627_1906    0    0    1    0    0
#&gt; ARS-BFGL-NGS-98142                 1    0    2    0    0</code></pre>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="gwas-ì ˆì°¨.html#cb30-1" tabindex="-1"></a><span class="fu">head</span>(bim)</span></code></pre></div>
<pre><code>#&gt; # A tibble: 6 Ã— 6
#&gt;   chr   id                               posg    pos alt   ref  
#&gt;   &lt;chr&gt; &lt;chr&gt;                           &lt;dbl&gt;  &lt;int&gt; &lt;chr&gt; &lt;chr&gt;
#&gt; 1 1     Hapmap43437-BTA-101873              0 135098 A     G    
#&gt; 2 1     ARS-BFGL-NGS-16466                  0 267940 A     G    
#&gt; 3 1     ARS-BFGL-NGS-105096                 0 353745 A     G    
#&gt; 4 1     Hapmap34944-BES1_Contig627_1906     0 393248 A     C    
#&gt; 5 1     ARS-BFGL-NGS-98142                  0 471078 A     G    
#&gt; 6 1     Hapmap53946-rs29015852              0 516404 A     G</code></pre>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="gwas-ì ˆì°¨.html#cb32-1" tabindex="-1"></a><span class="fu">head</span>(fam)</span></code></pre></div>
<pre><code>#&gt; # A tibble: 6 Ã— 6
#&gt;   fam   id    pat   mat     sex pheno
#&gt;   &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;int&gt; &lt;dbl&gt;
#&gt; 1 1     3530  0     0         0    -9
#&gt; 2 2     3528  0     0         0    -9
#&gt; 3 3     3527  0     0         0    -9
#&gt; 4 4     3195  0     0         0    -9
#&gt; 5 5     3994  0     0         0    -9
#&gt; 6 6     3786  0     0         0    -9</code></pre>
</div>
<div id="í‘œí˜„í˜•-ë°ì´í„°-ë¡œë”©-ì •ë ¬" class="section level2 hasAnchor" number="3.4">
<h2><span class="header-section-number">3.4</span> í‘œí˜„í˜• ë°ì´í„° ë¡œë”© &amp; ì •ë ¬<a href="#%ED%91%9C%ED%98%84%ED%98%95-%EB%8D%B0%EC%9D%B4%ED%84%B0-%EB%A1%9C%EB%94%A9-%EC%A0%95%EB%A0%AC" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="gwas-ì ˆì°¨.html#cb34-1" tabindex="-1"></a>pheno <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(pheno_file, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb34-2"><a href="gwas-ì ˆì°¨.html#cb34-2" tabindex="-1"></a></span>
<span id="cb34-3"><a href="gwas-ì ˆì°¨.html#cb34-3" tabindex="-1"></a>sample_id_chr <span class="ot">&lt;-</span> <span class="fu">colnames</span>(X)  <span class="co"># PLINK ì—´ì´ ìƒ˜í”Œ ID</span></span>
<span id="cb34-4"><a href="gwas-ì ˆì°¨.html#cb34-4" tabindex="-1"></a>scanID <span class="ot">&lt;-</span> <span class="fu">seq_along</span>(sample_id_chr)</span>
<span id="cb34-5"><a href="gwas-ì ˆì°¨.html#cb34-5" tabindex="-1"></a></span>
<span id="cb34-6"><a href="gwas-ì ˆì°¨.html#cb34-6" tabindex="-1"></a>pheno_ord <span class="ot">&lt;-</span> pheno[<span class="fu">match</span>(sample_id_chr, pheno<span class="sc">$</span>individual_id), , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb34-7"><a href="gwas-ì ˆì°¨.html#cb34-7" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="sc">!</span><span class="fu">any</span>(<span class="fu">is.na</span>(pheno_ord<span class="sc">$</span>individual_id)))</span>
<span id="cb34-8"><a href="gwas-ì ˆì°¨.html#cb34-8" tabindex="-1"></a>pheno_ord<span class="sc">$</span>sample_id <span class="ot">&lt;-</span> sample_id_chr</span>
<span id="cb34-9"><a href="gwas-ì ˆì°¨.html#cb34-9" tabindex="-1"></a>pheno_ord<span class="sc">$</span>scanID    <span class="ot">&lt;-</span> scanID</span>
<span id="cb34-10"><a href="gwas-ì ˆì°¨.html#cb34-10" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">identical</span>(<span class="fu">as.character</span>(pheno_ord<span class="sc">$</span>individual_id), pheno_ord<span class="sc">$</span>sample_id))</span>
<span id="cb34-11"><a href="gwas-ì ˆì°¨.html#cb34-11" tabindex="-1"></a><span class="co"># ë¯¸ë¦¬ë³´ê¸°</span></span>
<span id="cb34-12"><a href="gwas-ì ˆì°¨.html#cb34-12" tabindex="-1"></a><span class="fu">str</span>(pheno_ord)</span></code></pre></div>
<pre><code>#&gt; tibble [200 Ã— 15] (S3: tbl_df/tbl/data.frame)
#&gt;  $ individual_id              : num [1:200] 3530 3528 3527 3195 3994 ...
#&gt;  $ breed                      : chr [1:200] &quot;holstein&quot; &quot;holstein&quot; &quot;holstein&quot; &quot;holstein&quot; ...
#&gt;  $ sire_id                    : num [1:200] 3074 2930 3048 3056 3050 ...
#&gt;  $ dam_id                     : num [1:200] 3210 3343 3281 3016 3799 ...
#&gt;  $ birth_date                 : Date[1:200], format: &quot;2009-03-02&quot; &quot;2009-01-17&quot; ...
#&gt;  $ HYS                        : num [1:200] 1.16e+10 1.18e+10 1.18e+10 1.18e+10 1.16e+10 ...
#&gt;  $ farm_id                    : num [1:200] 116482 117952 117952 117952 116482 ...
#&gt;  $ parity                     : num [1:200] 1 1 1 1 1 1 1 1 1 1 ...
#&gt;  $ days_in_milk               : num [1:200] 482 531 382 556 264 447 425 400 345 385 ...
#&gt;  $ milk_yield_305.days        : num [1:200] 7965 9717 12554 9951 7982 ...
#&gt;  $ milk_fat_yield_305.days    : num [1:200] 281 314 468 445 271 297 305 347 359 355 ...
#&gt;  $ milk_protein_yield_305.days: num [1:200] 260 302 384 334 260 240 230 327 293 309 ...
#&gt;  $ somatic_cell_count         : num [1:200] 1.39 1.89 2.14 4.31 3.9 ...
#&gt;  $ sample_id                  : chr [1:200] &quot;3530&quot; &quot;3528&quot; &quot;3527&quot; &quot;3195&quot; ...
#&gt;  $ scanID                     : int [1:200] 1 2 3 4 5 6 7 8 9 10 ...</code></pre>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="gwas-ì ˆì°¨.html#cb36-1" tabindex="-1"></a><span class="co"># í‘œí˜„í˜• ë¶„í¬ í™•ì¸</span></span>
<span id="cb36-2"><a href="gwas-ì ˆì°¨.html#cb36-2" tabindex="-1"></a><span class="co"># --- ë¶„ì„í•  í˜•ì§ˆì„ ì´ ë³€ìˆ˜ì— ì§€ì • ---</span></span>
<span id="cb36-3"><a href="gwas-ì ˆì°¨.html#cb36-3" tabindex="-1"></a>single_trait <span class="ot">&lt;-</span> <span class="st">&quot;days_in_milk&quot;</span></span>
<span id="cb36-4"><a href="gwas-ì ˆì°¨.html#cb36-4" tabindex="-1"></a></span>
<span id="cb36-5"><a href="gwas-ì ˆì°¨.html#cb36-5" tabindex="-1"></a>p<span class="ot">=</span><span class="fu">ggplot</span>(pheno_ord, <span class="fu">aes</span>(<span class="at">x =</span> .data[[single_trait]])) <span class="sc">+</span></span>
<span id="cb36-6"><a href="gwas-ì ˆì°¨.html#cb36-6" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">y =</span> <span class="fu">after_stat</span>(density)), <span class="at">bins =</span> <span class="dv">30</span>, <span class="at">fill =</span> <span class="st">&quot;grey70&quot;</span>, <span class="at">color =</span> <span class="st">&quot;white&quot;</span>) <span class="sc">+</span></span>
<span id="cb36-7"><a href="gwas-ì ˆì°¨.html#cb36-7" tabindex="-1"></a>  <span class="fu">geom_density</span>(<span class="at">linewidth =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb36-8"><a href="gwas-ì ˆì°¨.html#cb36-8" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb36-9"><a href="gwas-ì ˆì°¨.html#cb36-9" tabindex="-1"></a>    <span class="at">title =</span> <span class="fu">paste</span>(<span class="st">&quot;Distribution of&quot;</span>, single_trait),</span>
<span id="cb36-10"><a href="gwas-ì ˆì°¨.html#cb36-10" tabindex="-1"></a>    <span class="at">x =</span> single_trait, </span>
<span id="cb36-11"><a href="gwas-ì ˆì°¨.html#cb36-11" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">&quot;Density&quot;</span></span>
<span id="cb36-12"><a href="gwas-ì ˆì°¨.html#cb36-12" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb36-13"><a href="gwas-ì ˆì°¨.html#cb36-13" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb36-14"><a href="gwas-ì ˆì°¨.html#cb36-14" tabindex="-1"></a></span>
<span id="cb36-15"><a href="gwas-ì ˆì°¨.html#cb36-15" tabindex="-1"></a><span class="co"># ë³€ìˆ˜ì— ì €ì¥ëœ í”Œë¡¯ í™•ì¸</span></span>
<span id="cb36-16"><a href="gwas-ì ˆì°¨.html#cb36-16" tabindex="-1"></a><span class="fu">print</span>(p)</span></code></pre></div>
<p><img src="my-final-book_files/figure-html/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="gwas-ì ˆì°¨.html#cb37-1" tabindex="-1"></a><span class="co"># ggsave() í•¨ìˆ˜ë¥¼ ì‚¬ìš©í•˜ì—¬ í”Œë¡¯ì„ png íŒŒì¼ë¡œ ì €ì¥</span></span>
<span id="cb37-2"><a href="gwas-ì ˆì°¨.html#cb37-2" tabindex="-1"></a><span class="fu">ggsave</span>(</span>
<span id="cb37-3"><a href="gwas-ì ˆì°¨.html#cb37-3" tabindex="-1"></a>  <span class="at">filename =</span> <span class="fu">file.path</span>(figs_dir, <span class="fu">paste0</span>(<span class="st">&quot;pheno_distribution_&quot;</span>, single_trait, <span class="st">&quot;.png&quot;</span>)),</span>
<span id="cb37-4"><a href="gwas-ì ˆì°¨.html#cb37-4" tabindex="-1"></a>  <span class="at">plot =</span> p,</span>
<span id="cb37-5"><a href="gwas-ì ˆì°¨.html#cb37-5" tabindex="-1"></a>  <span class="at">width =</span> <span class="dv">6</span>,</span>
<span id="cb37-6"><a href="gwas-ì ˆì°¨.html#cb37-6" tabindex="-1"></a>  <span class="at">height =</span> <span class="dv">4</span>,</span>
<span id="cb37-7"><a href="gwas-ì ˆì°¨.html#cb37-7" tabindex="-1"></a>  <span class="at">dpi =</span> <span class="dv">300</span></span>
<span id="cb37-8"><a href="gwas-ì ˆì°¨.html#cb37-8" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="grmìœ ì „ì²´-ê´€ê³„-í–‰ë ¬-ê³„ì‚°-ê°„ë‹¨-ì‹œê°í™”" class="section level2 hasAnchor" number="3.5">
<h2><span class="header-section-number">3.5</span> GRM(ìœ ì „ì²´ ê´€ê³„ í–‰ë ¬) ê³„ì‚° &amp; ê°„ë‹¨ ì‹œê°í™”<a href="#grm%EC%9C%A0%EC%A0%84%EC%B2%B4-%EA%B4%80%EA%B3%84-%ED%96%89%EB%A0%AC-%EA%B3%84%EC%82%B0-%EA%B0%84%EB%8B%A8-%EC%8B%9C%EA%B0%81%ED%99%94" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="gwas-ì ˆì°¨.html#cb38-1" tabindex="-1"></a><span class="co"># A.matì€ -1/0/1 ì½”ë”©ëœ genotype matrix ì´ìš©</span></span>
<span id="cb38-2"><a href="gwas-ì ˆì°¨.html#cb38-2" tabindex="-1"></a>G_mat <span class="ot">&lt;-</span> <span class="fu">t</span>(X)              <span class="co"># ìƒ˜í”Œ x SNP</span></span>
<span id="cb38-3"><a href="gwas-ì ˆì°¨.html#cb38-3" tabindex="-1"></a><span class="fu">colnames</span>(G_mat) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(X)</span>
<span id="cb38-4"><a href="gwas-ì ˆì°¨.html#cb38-4" tabindex="-1"></a><span class="fu">rownames</span>(G_mat) <span class="ot">&lt;-</span> sample_id_chr</span>
<span id="cb38-5"><a href="gwas-ì ˆì°¨.html#cb38-5" tabindex="-1"></a></span>
<span id="cb38-6"><a href="gwas-ì ˆì°¨.html#cb38-6" tabindex="-1"></a>G_code <span class="ot">&lt;-</span> G_mat <span class="sc">-</span> <span class="dv">1</span>        <span class="co"># 0/1/2 -&gt; -1/0/1</span></span>
<span id="cb38-7"><a href="gwas-ì ˆì°¨.html#cb38-7" tabindex="-1"></a>KI     <span class="ot">&lt;-</span> sommer<span class="sc">::</span><span class="fu">A.mat</span>(G_code)</span></code></pre></div>
<pre><code>#&gt; Imputing markers with mean value</code></pre>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="gwas-ì ˆì°¨.html#cb40-1" tabindex="-1"></a><span class="co"># ìˆ˜ì¹˜ì•ˆì •ì„±(ì•„ì£¼ ì‘ì€ ridge)</span></span>
<span id="cb40-2"><a href="gwas-ì ˆì°¨.html#cb40-2" tabindex="-1"></a><span class="fu">diag</span>(KI) <span class="ot">&lt;-</span> <span class="fu">diag</span>(KI) <span class="sc">+</span> <span class="fl">1e-4</span></span>
<span id="cb40-3"><a href="gwas-ì ˆì°¨.html#cb40-3" tabindex="-1"></a></span>
<span id="cb40-4"><a href="gwas-ì ˆì°¨.html#cb40-4" tabindex="-1"></a><span class="co"># scanID ê¸°ì¤€ì˜ dimnames ì§€ì •</span></span>
<span id="cb40-5"><a href="gwas-ì ˆì°¨.html#cb40-5" tabindex="-1"></a><span class="fu">rownames</span>(KI) <span class="ot">&lt;-</span> <span class="fu">as.character</span>(scanID)</span>
<span id="cb40-6"><a href="gwas-ì ˆì°¨.html#cb40-6" tabindex="-1"></a><span class="fu">colnames</span>(KI) <span class="ot">&lt;-</span> <span class="fu">as.character</span>(scanID)</span>
<span id="cb40-7"><a href="gwas-ì ˆì°¨.html#cb40-7" tabindex="-1"></a></span>
<span id="cb40-8"><a href="gwas-ì ˆì°¨.html#cb40-8" tabindex="-1"></a><span class="fu">stopifnot</span>(</span>
<span id="cb40-9"><a href="gwas-ì ˆì°¨.html#cb40-9" tabindex="-1"></a>  <span class="fu">identical</span>(<span class="fu">rownames</span>(KI), <span class="fu">as.character</span>(scanID)),</span>
<span id="cb40-10"><a href="gwas-ì ˆì°¨.html#cb40-10" tabindex="-1"></a>  <span class="fu">identical</span>(<span class="fu">colnames</span>(KI), <span class="fu">as.character</span>(scanID))</span>
<span id="cb40-11"><a href="gwas-ì ˆì°¨.html#cb40-11" tabindex="-1"></a>)</span>
<span id="cb40-12"><a href="gwas-ì ˆì°¨.html#cb40-12" tabindex="-1"></a></span>
<span id="cb40-13"><a href="gwas-ì ˆì°¨.html#cb40-13" tabindex="-1"></a><span class="co"># ìƒìœ„ 5x5 í™•ì¸</span></span>
<span id="cb40-14"><a href="gwas-ì ˆì°¨.html#cb40-14" tabindex="-1"></a>KI[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>#&gt;             1            2           3            4           5
#&gt; 1  1.01954033 -0.026190312 -0.01241066  0.090020422 -0.04370613
#&gt; 2 -0.02619031  0.974199397  0.07949861 -0.009271238  0.03287250
#&gt; 3 -0.01241066  0.079498610  0.93486718 -0.051821286  0.01365447
#&gt; 4  0.09002042 -0.009271238 -0.05182129  0.997304816  0.04189677
#&gt; 5 -0.04370613  0.032872500  0.01365447  0.041896774  1.09127396</code></pre>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="gwas-ì ˆì°¨.html#cb42-1" tabindex="-1"></a><span class="co"># íˆíŠ¸ë§µ</span></span>
<span id="cb42-2"><a href="gwas-ì ˆì°¨.html#cb42-2" tabindex="-1"></a>colfunc <span class="ot">&lt;-</span> grDevices<span class="sc">::</span><span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">&quot;steelblue4&quot;</span>,<span class="st">&quot;springgreen&quot;</span>,<span class="st">&quot;yellow&quot;</span>))</span>
<span id="cb42-3"><a href="gwas-ì ˆì°¨.html#cb42-3" tabindex="-1"></a></span>
<span id="cb42-4"><a href="gwas-ì ˆì°¨.html#cb42-4" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">file.path</span>(report_dir, <span class="st">&quot;A_matrix_heatmap.png&quot;</span>), <span class="at">width=</span><span class="dv">2000</span>, <span class="at">height=</span><span class="dv">1400</span>, <span class="at">res=</span><span class="dv">200</span>)</span>
<span id="cb42-5"><a href="gwas-ì ˆì°¨.html#cb42-5" tabindex="-1"></a>stats<span class="sc">::</span><span class="fu">heatmap</span>(KI, <span class="at">col =</span> <span class="fu">colfunc</span>(<span class="dv">100</span>), <span class="at">Colv =</span> <span class="st">&quot;Rowv&quot;</span>, <span class="at">symmetric =</span> <span class="cn">TRUE</span>)</span>
<span id="cb42-6"><a href="gwas-ì ˆì°¨.html#cb42-6" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<pre><code>#&gt; png 
#&gt;   2</code></pre>
<p>
<em>GRM íˆíŠ¸ë§µ</em>
</p>
<p><img src="05_gwas/holstein/A_matrix_heatmap.png" width="1000" style="display: block; margin: auto;" /></p>
</div>
<div id="gwas-ë¶„ì„ìš©-ë°ì´í„°-ê°ì²´-ë§Œë“¤ê¸°" class="section level2 hasAnchor" number="3.6">
<h2><span class="header-section-number">3.6</span> GWAS ë¶„ì„ìš© ë°ì´í„° ê°ì²´ ë§Œë“¤ê¸°<a href="#gwas-%EB%B6%84%EC%84%9D%EC%9A%A9-%EB%8D%B0%EC%9D%B4%ED%84%B0-%EA%B0%9D%EC%B2%B4-%EB%A7%8C%EB%93%A4%EA%B8%B0" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="gwas-ì ˆì°¨.html#cb44-1" tabindex="-1"></a>bim <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="at">index =</span> <span class="fu">seq_len</span>(<span class="fu">nrow</span>(bim)), bim)</span>
<span id="cb44-2"><a href="gwas-ì ˆì°¨.html#cb44-2" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">all.equal</span>(<span class="fu">rownames</span>(X), bim<span class="sc">$</span>id))</span>
<span id="cb44-3"><a href="gwas-ì ˆì°¨.html#cb44-3" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">all.equal</span>(<span class="fu">colnames</span>(X), pheno_ord<span class="sc">$</span>sample_id))</span>
<span id="cb44-4"><a href="gwas-ì ˆì°¨.html#cb44-4" tabindex="-1"></a></span>
<span id="cb44-5"><a href="gwas-ì ˆì°¨.html#cb44-5" tabindex="-1"></a><span class="fu">rownames</span>(X) <span class="ot">&lt;-</span> bim<span class="sc">$</span>index</span>
<span id="cb44-6"><a href="gwas-ì ˆì°¨.html#cb44-6" tabindex="-1"></a><span class="fu">colnames</span>(X) <span class="ot">&lt;-</span> scanID</span>
<span id="cb44-7"><a href="gwas-ì ˆì°¨.html#cb44-7" tabindex="-1"></a></span>
<span id="cb44-8"><a href="gwas-ì ˆì°¨.html#cb44-8" tabindex="-1"></a><span class="co"># GENESIS íŒ¨í‚¤ì§€ì—ì„œ ìš”êµ¬í•˜ëŠ” ë°ì´í„° ê°ì²´ ë§Œë“¤ê¸°</span></span>
<span id="cb44-9"><a href="gwas-ì ˆì°¨.html#cb44-9" tabindex="-1"></a>mg <span class="ot">&lt;-</span> GWASTools<span class="sc">::</span><span class="fu">MatrixGenotypeReader</span>(</span>
<span id="cb44-10"><a href="gwas-ì ˆì°¨.html#cb44-10" tabindex="-1"></a>  <span class="at">genotype   =</span> X,</span>
<span id="cb44-11"><a href="gwas-ì ˆì°¨.html#cb44-11" tabindex="-1"></a>  <span class="at">snpID      =</span> bim<span class="sc">$</span>index,</span>
<span id="cb44-12"><a href="gwas-ì ˆì°¨.html#cb44-12" tabindex="-1"></a>  <span class="at">chromosome =</span> <span class="fu">as.integer</span>(bim<span class="sc">$</span>chr),</span>
<span id="cb44-13"><a href="gwas-ì ˆì°¨.html#cb44-13" tabindex="-1"></a>  <span class="at">position   =</span> <span class="fu">as.integer</span>(bim<span class="sc">$</span>pos),</span>
<span id="cb44-14"><a href="gwas-ì ˆì°¨.html#cb44-14" tabindex="-1"></a>  <span class="at">scanID     =</span> scanID,</span>
<span id="cb44-15"><a href="gwas-ì ˆì°¨.html#cb44-15" tabindex="-1"></a>  <span class="at">autosomeCode =</span> <span class="dv">1</span><span class="dt">L</span><span class="sc">:</span><span class="dv">29</span><span class="dt">L</span>,</span>
<span id="cb44-16"><a href="gwas-ì ˆì°¨.html#cb44-16" tabindex="-1"></a>  <span class="at">XchromCode   =</span> <span class="dv">30</span><span class="dt">L</span>, <span class="at">YchromCode =</span> <span class="dv">31</span><span class="dt">L</span>, <span class="at">XYchromCode =</span> <span class="dv">32</span><span class="dt">L</span>, <span class="at">MchromCode =</span> <span class="dv">33</span><span class="dt">L</span></span>
<span id="cb44-17"><a href="gwas-ì ˆì°¨.html#cb44-17" tabindex="-1"></a>)</span>
<span id="cb44-18"><a href="gwas-ì ˆì°¨.html#cb44-18" tabindex="-1"></a><span class="co"># ìœ ì „ìí˜• ë°ì´í„° ê°ì²´ ìƒì„±</span></span>
<span id="cb44-19"><a href="gwas-ì ˆì°¨.html#cb44-19" tabindex="-1"></a>genoData <span class="ot">&lt;-</span> GWASTools<span class="sc">::</span><span class="fu">GenotypeData</span>(mg)</span>
<span id="cb44-20"><a href="gwas-ì ˆì°¨.html#cb44-20" tabindex="-1"></a><span class="co"># í‘œí˜„í˜• ë°ì´í„° ê°ì²´ ìƒì„±</span></span>
<span id="cb44-21"><a href="gwas-ì ˆì°¨.html#cb44-21" tabindex="-1"></a>scanAnno <span class="ot">&lt;-</span> GWASTools<span class="sc">::</span><span class="fu">ScanAnnotationDataFrame</span>(pheno_ord)</span>
<span id="cb44-22"><a href="gwas-ì ˆì°¨.html#cb44-22" tabindex="-1"></a><span class="co"># ê°ì²´ êµ¬ì¡° í™•ì¸</span></span>
<span id="cb44-23"><a href="gwas-ì ˆì°¨.html#cb44-23" tabindex="-1"></a><span class="fu">str</span>(genoData)</span></code></pre></div>
<pre><code>#&gt; Formal class &#39;GenotypeData&#39; [package &quot;GWASTools&quot;] with 3 slots
#&gt;   ..@ data     :Formal class &#39;MatrixGenotypeReader&#39; [package &quot;GWASTools&quot;] with 10 slots
#&gt;   .. .. ..@ snpID       : int [1:43686] 1 2 3 4 5 6 7 8 9 10 ...
#&gt;   .. .. ..@ chromosome  : int [1:43686] 1 1 1 1 1 1 1 1 1 1 ...
#&gt;   .. .. ..@ position    : int [1:43686] 135098 267940 353745 393248 471078 516404 845494 883895 929617 950841 ...
#&gt;   .. .. ..@ scanID      : int [1:200] 1 2 3 4 5 6 7 8 9 10 ...
#&gt;   .. .. ..@ genotype    : int [1:43686, 1:200] 1 0 1 0 1 1 0 0 0 0 ...
#&gt;   .. .. .. ..- attr(*, &quot;dimnames&quot;)=List of 2
#&gt;   .. .. .. .. ..$ : chr [1:43686] &quot;1&quot; &quot;2&quot; &quot;3&quot; &quot;4&quot; ...
#&gt;   .. .. .. .. ..$ : chr [1:200] &quot;1&quot; &quot;2&quot; &quot;3&quot; &quot;4&quot; ...
#&gt;   .. .. ..@ autosomeCode: int [1:29] 1 2 3 4 5 6 7 8 9 10 ...
#&gt;   .. .. ..@ XchromCode  : int 30
#&gt;   .. .. ..@ YchromCode  : int 31
#&gt;   .. .. ..@ XYchromCode : int 32
#&gt;   .. .. ..@ MchromCode  : int 33
#&gt;   ..@ snpAnnot : NULL
#&gt;   ..@ scanAnnot: NULL</code></pre>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="gwas-ì ˆì°¨.html#cb46-1" tabindex="-1"></a><span class="fu">str</span>(scanAnno<span class="sc">@</span>data)   </span></code></pre></div>
<pre><code>#&gt; tibble [200 Ã— 15] (S3: tbl_df/tbl/data.frame)
#&gt;  $ individual_id              : num [1:200] 3530 3528 3527 3195 3994 ...
#&gt;  $ breed                      : chr [1:200] &quot;holstein&quot; &quot;holstein&quot; &quot;holstein&quot; &quot;holstein&quot; ...
#&gt;  $ sire_id                    : num [1:200] 3074 2930 3048 3056 3050 ...
#&gt;  $ dam_id                     : num [1:200] 3210 3343 3281 3016 3799 ...
#&gt;  $ birth_date                 : Date[1:200], format: &quot;2009-03-02&quot; &quot;2009-01-17&quot; ...
#&gt;  $ HYS                        : num [1:200] 1.16e+10 1.18e+10 1.18e+10 1.18e+10 1.16e+10 ...
#&gt;  $ farm_id                    : num [1:200] 116482 117952 117952 117952 116482 ...
#&gt;  $ parity                     : num [1:200] 1 1 1 1 1 1 1 1 1 1 ...
#&gt;  $ days_in_milk               : num [1:200] 482 531 382 556 264 447 425 400 345 385 ...
#&gt;  $ milk_yield_305.days        : num [1:200] 7965 9717 12554 9951 7982 ...
#&gt;  $ milk_fat_yield_305.days    : num [1:200] 281 314 468 445 271 297 305 347 359 355 ...
#&gt;  $ milk_protein_yield_305.days: num [1:200] 260 302 384 334 260 240 230 327 293 309 ...
#&gt;  $ somatic_cell_count         : num [1:200] 1.39 1.89 2.14 4.31 3.9 ...
#&gt;  $ sample_id                  : chr [1:200] &quot;3530&quot; &quot;3528&quot; &quot;3527&quot; &quot;3195&quot; ...
#&gt;  $ scanID                     : int [1:200] 1 2 3 4 5 6 7 8 9 10 ...</code></pre>
</div>
<div id="gwas-ê·€ë¬´ëª¨í˜•-ì í•©-ë°-ìœ ì „ë ¥snp-heritability-ê³„ì‚°" class="section level2 hasAnchor" number="3.7">
<h2><span class="header-section-number">3.7</span> GWAS ê·€ë¬´ëª¨í˜• ì í•© ë° ìœ ì „ë ¥(SNP heritability) ê³„ì‚°<a href="#gwas-%EA%B7%80%EB%AC%B4%EB%AA%A8%ED%98%95-%EC%A0%81%ED%95%A9-%EB%B0%8F-%EC%9C%A0%EC%A0%84%EB%A0%A5snp-heritability-%EA%B3%84%EC%82%B0" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="gwas-ì ˆì°¨.html#cb48-1" tabindex="-1"></a><span class="co"># --- ë¶„ì„í•  í˜•ì§ˆì„ ì´ ë³€ìˆ˜ì— ì§€ì • ---</span></span>
<span id="cb48-2"><a href="gwas-ì ˆì°¨.html#cb48-2" tabindex="-1"></a>single_trait <span class="ot">&lt;-</span> <span class="st">&quot;days_in_milk&quot;</span></span>
<span id="cb48-3"><a href="gwas-ì ˆì°¨.html#cb48-3" tabindex="-1"></a></span>
<span id="cb48-4"><a href="gwas-ì ˆì°¨.html#cb48-4" tabindex="-1"></a><span class="co"># ë¶„ì„í•  í˜•ì§ˆì´ ë°ì´í„°ì— ì¡´ì¬í•˜ëŠ”ì§€ í™•ì¸. ì—†ìœ¼ë©´ ì—ëŸ¬ë¥¼ ë°œìƒì‹œí‚¤ê³  ì¤‘ë‹¨</span></span>
<span id="cb48-5"><a href="gwas-ì ˆì°¨.html#cb48-5" tabindex="-1"></a><span class="fu">stopifnot</span>(single_trait <span class="sc">%in%</span> <span class="fu">colnames</span>(scanAnno<span class="sc">@</span>data))</span>
<span id="cb48-6"><a href="gwas-ì ˆì°¨.html#cb48-6" tabindex="-1"></a></span>
<span id="cb48-7"><a href="gwas-ì ˆì°¨.html#cb48-7" tabindex="-1"></a><span class="co"># --- ë°ì´í„° ì „ì²˜ë¦¬ ---</span></span>
<span id="cb48-8"><a href="gwas-ì ˆì°¨.html#cb48-8" tabindex="-1"></a></span>
<span id="cb48-9"><a href="gwas-ì ˆì°¨.html#cb48-9" tabindex="-1"></a><span class="co"># 1. ëª¨ë¸ì—ì„œ íš¨ê³¼ë¥¼ ë¶„ì„í•  ë³€ìˆ˜ë“¤ì„ factor(ë²”ì£¼í˜• ë³€ìˆ˜)ë¡œ ë³€í™˜</span></span>
<span id="cb48-10"><a href="gwas-ì ˆì°¨.html#cb48-10" tabindex="-1"></a>pheno_ord<span class="sc">$</span>farm_id <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(pheno_ord<span class="sc">$</span>farm_id)</span>
<span id="cb48-11"><a href="gwas-ì ˆì°¨.html#cb48-11" tabindex="-1"></a>pheno_ord<span class="sc">$</span>scanID <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(pheno_ord<span class="sc">$</span>scanID)</span>
<span id="cb48-12"><a href="gwas-ì ˆì°¨.html#cb48-12" tabindex="-1"></a></span>
<span id="cb48-13"><a href="gwas-ì ˆì°¨.html#cb48-13" tabindex="-1"></a><span class="co"># 2. ë¶„ì„í•  í˜•ì§ˆ(days_in_milk)ì— ê²°ì¸¡ì¹˜(NA)ê°€ ìˆëŠ” ìƒ˜í”Œì„ ì œê±°í•˜ì—¬</span></span>
<span id="cb48-14"><a href="gwas-ì ˆì°¨.html#cb48-14" tabindex="-1"></a><span class="co">#    ìµœì¢… ë¶„ì„ ë°ì´í„°ì…‹ &#39;PH&#39;ë¥¼ ìƒì„±</span></span>
<span id="cb48-15"><a href="gwas-ì ˆì°¨.html#cb48-15" tabindex="-1"></a><span class="co">#    droplevels() í•¨ìˆ˜ëŠ” PH ë°ì´í„°ì…‹ì— ë” ì´ìƒ ì¡´ì¬í•˜ì§€ ì•ŠëŠ” ìƒ˜í”Œì´ë‚˜ ë†ì¥ì˜ ë ˆë²¨ ì •ë³´ë¥¼ ì œê±°</span></span>
<span id="cb48-16"><a href="gwas-ì ˆì°¨.html#cb48-16" tabindex="-1"></a>PH <span class="ot">&lt;-</span> <span class="fu">droplevels</span>(<span class="fu">subset</span>(pheno_ord, <span class="sc">!</span><span class="fu">is.na</span>(days_in_milk)))</span>
<span id="cb48-17"><a href="gwas-ì ˆì°¨.html#cb48-17" tabindex="-1"></a></span>
<span id="cb48-18"><a href="gwas-ì ˆì°¨.html#cb48-18" tabindex="-1"></a><span class="co"># 3. Kinship í–‰ë ¬(KI)ì„ ìµœì¢… ë¶„ì„ ë°ì´í„°ì…‹ &#39;PH&#39;ì— ë§ì¶° ìë¥¸ë‹¤.</span></span>
<span id="cb48-19"><a href="gwas-ì ˆì°¨.html#cb48-19" tabindex="-1"></a><span class="co">#    ë°˜ë“œì‹œ &#39;PH&#39;ì˜ ìƒ˜í”Œ ëª©ë¡ê³¼ ìˆœì„œë¥¼ ê¸°ì¤€ìœ¼ë¡œ ì˜ë¼ì•¼,</span></span>
<span id="cb48-20"><a href="gwas-ì ˆì°¨.html#cb48-20" tabindex="-1"></a><span class="co">#    í‘œí˜„í˜• ë°ì´í„°ì™€ Kinship ë°ì´í„°ì˜ ìƒ˜í”Œì´ 1:1ë¡œ ì •í™•í•˜ê²Œ ì¼ì¹˜í•œë‹¤.</span></span>
<span id="cb48-21"><a href="gwas-ì ˆì°¨.html#cb48-21" tabindex="-1"></a>K_scan <span class="ot">&lt;-</span> KI[<span class="fu">levels</span>(PH<span class="sc">$</span>scanID), <span class="fu">levels</span>(PH<span class="sc">$</span>scanID)]</span>
<span id="cb48-22"><a href="gwas-ì ˆì°¨.html#cb48-22" tabindex="-1"></a></span>
<span id="cb48-23"><a href="gwas-ì ˆì°¨.html#cb48-23" tabindex="-1"></a><span class="co"># ë†ì¥ íš¨ê³¼ì™€ ê°œì²´ë³„ ìœ ì „ íš¨ê³¼ë¥¼ ì„ì˜ íš¨ê³¼(random effect)ë¡œ ì„¤ì •í•˜ì—¬ ëª¨ë¸ì„ ë§Œë“ ë‹¤.</span></span>
<span id="cb48-24"><a href="gwas-ì ˆì°¨.html#cb48-24" tabindex="-1"></a><span class="co"># lmebreed í•¨ìˆ˜ëŠ” í˜ˆì—°ê´€ê³„(kinship)ë¥¼ ê³ ë ¤í•œ í˜¼í•© ëª¨ë¸ì„ ì í•©í•´ì¤€ë‹¤.</span></span>
<span id="cb48-25"><a href="gwas-ì ˆì°¨.html#cb48-25" tabindex="-1"></a>mix <span class="ot">&lt;-</span> lme4breeding<span class="sc">::</span><span class="fu">lmebreed</span>(</span>
<span id="cb48-26"><a href="gwas-ì ˆì°¨.html#cb48-26" tabindex="-1"></a>  <span class="co"># ëª¨ë¸ ê³µì‹: days_in_milkë¥¼ ê°œì²´ íš¨ê³¼(scanID)ì™€ ë†ì¥ íš¨ê³¼(farm_id)ë¡œ ì„¤ëª…</span></span>
<span id="cb48-27"><a href="gwas-ì ˆì°¨.html#cb48-27" tabindex="-1"></a>  days_in_milk <span class="sc">~</span> (<span class="dv">1</span><span class="sc">|</span>scanID) <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>farm_id),</span>
<span id="cb48-28"><a href="gwas-ì ˆì°¨.html#cb48-28" tabindex="-1"></a>  </span>
<span id="cb48-29"><a href="gwas-ì ˆì°¨.html#cb48-29" tabindex="-1"></a>  <span class="co"># relmat ì¸ì: scanIDì˜ ì„ì˜ íš¨ê³¼ëŠ” K_scan í–‰ë ¬ì— ì •ì˜ëœ ìœ ì „ì  ê´€ê³„ë¥¼ ë”°ë¥´ë„ë¡ ì§€ì •í•œë‹¤.</span></span>
<span id="cb48-30"><a href="gwas-ì ˆì°¨.html#cb48-30" tabindex="-1"></a>  <span class="at">relmat  =</span> <span class="fu">list</span>(<span class="at">scanID =</span> K_scan),</span>
<span id="cb48-31"><a href="gwas-ì ˆì°¨.html#cb48-31" tabindex="-1"></a>  </span>
<span id="cb48-32"><a href="gwas-ì ˆì°¨.html#cb48-32" tabindex="-1"></a>  <span class="co"># verbose=TRUE: ëª¨ë¸ ê³„ì‚° ê³¼ì •ì„ í™”ë©´ì— ì¶œë ¥</span></span>
<span id="cb48-33"><a href="gwas-ì ˆì°¨.html#cb48-33" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="cn">TRUE</span>,</span>
<span id="cb48-34"><a href="gwas-ì ˆì°¨.html#cb48-34" tabindex="-1"></a>  </span>
<span id="cb48-35"><a href="gwas-ì ˆì°¨.html#cb48-35" tabindex="-1"></a>  <span class="co"># data ì¸ì: ëª¨ë¸ì— ì‚¬ìš©í•  ë°ì´í„°ì…‹ìœ¼ë¡œ PHë¥¼ ì§€ì •</span></span>
<span id="cb48-36"><a href="gwas-ì ˆì°¨.html#cb48-36" tabindex="-1"></a>  <span class="at">data    =</span> PH</span>
<span id="cb48-37"><a href="gwas-ì ˆì°¨.html#cb48-37" tabindex="-1"></a>)</span>
<span id="cb48-38"><a href="gwas-ì ˆì°¨.html#cb48-38" tabindex="-1"></a></span>
<span id="cb48-39"><a href="gwas-ì ˆì°¨.html#cb48-39" tabindex="-1"></a><span class="co"># ëª¨ë¸ ë¶„ì„ ê²°ê³¼ ìš”ì•½ë³¸ ì¶œë ¥</span></span>
<span id="cb48-40"><a href="gwas-ì ˆì°¨.html#cb48-40" tabindex="-1"></a><span class="fu">summary</span>(mix)</span></code></pre></div>
<pre><code>#&gt; Linear mixed model fit by REML [&#39;lmerlmebreed&#39;]
#&gt; Formula: days_in_milk ~ (1 | scanID) + (1 | farm_id)
#&gt;    Data: PH
#&gt; 
#&gt; REML criterion at convergence: 2551.3
#&gt; 
#&gt; Scaled residuals: 
#&gt;     Min      1Q  Median      3Q     Max 
#&gt; -1.7342 -0.4654 -0.2040  0.3307  7.6392 
#&gt; 
#&gt; Random effects:
#&gt;  Groups   Name        Variance Std.Dev.
#&gt;  scanID   (Intercept)  3794.3   61.60  
#&gt;  farm_id  (Intercept)   289.4   17.01  
#&gt;  Residual             17284.7  131.47  
#&gt; Number of obs: 200, groups:  scanID, 200; farm_id, 20
#&gt; 
#&gt; Fixed effects:
#&gt;             Estimate Std. Error t value
#&gt; (Intercept)   360.25      10.71   33.65</code></pre>
<!-- ===== í—¤ë” ë°”ê°€ ìˆëŠ” ì¹´ë“œ ë²„ì „ ===== -->
<style>
.gwas-card{
  border:1px solid #e5e7eb; border-radius:12px; overflow:hidden; margin:18px 0;
  background:#fcfcff; box-shadow:0 1px 0 rgba(17,24,39,.04);
}
.gwas-header{
  padding:10px 16px; background:#e0e7ff; border-bottom:1px solid #c7d2fe;
  display:flex; align-items:center; gap:.6rem;
}
.gwas-header .title{ font-weight:700; }
.gwas-body{ padding:16px 18px; }

/* ë‚˜ë¨¸ì§€ ìŠ¤íƒ€ì¼ì€ ìœ„ì™€ ë™ì¼ */
.gwas-sub{ color:#4b5563; margin:0 0 .8rem; font-size:.95rem; }
.gwas-dl{ margin:0; }
.gwas-dl dt{ font-weight:700; margin-top:.6rem; }
.gwas-dl dd{ margin:.15rem 0 .5rem 0; color:#374151; }
.gwas-kpis{ display:flex; flex-wrap:wrap; gap:.5rem; margin:.6rem 0 .2rem; }
.kpi{ display:inline-flex; align-items:center; gap:.35rem;
      padding:.3rem .55rem; border-radius:999px; background:#c7d2fe; font-weight:600;
      border:1px solid #818cf8; color:#1f2937; }
.badge{ font-size:.8rem; padding:.05rem .35rem; border-radius:6px; background:#9ca3af; color:#fff; }
.gwas-grid{ display:grid; grid-template-columns: 1fr; gap:.6rem; }
@media(min-width:700px){ .gwas-grid{ grid-template-columns:1fr 1fr; } }
.gwas-table{ width:100%; border-collapse:collapse; font-size:.95rem; }
.gwas-table th, .gwas-table td{ padding:.4rem .55rem; border-bottom:1px solid #e5e7eb; text-align:left; }
.gwas-note{ margin-top:.6rem; color:#374151; }
.hr-soft{ border:0; border-top:1px dashed #e5e7eb; margin:1rem 0; }
</style>

<div class="gwas-card" id="null-model-summary">
  <div class="gwas-header">
    <div class="title icon-wrap"><span class="emoji">â“</span><span>Null model ìš”ì•½ â€” days_in_milk</span></div>
    <div class="meta" style="margin-left:auto; font-size:.9rem; color:#374151;">Vg/Vc/Ve & hÂ²</div>
  </div>
  <div class="gwas-body">
    <h4 style="margin:.2rem 0 .8rem;"> 1. ì„ì˜ íš¨ê³¼ (Random effects) ë¶„ì„</h4>
    <p class="gwas-sub">ì´ ë¶€ë¶„ì´ ê²°ê³¼ í•´ì„ì˜ í•µì‹¬ì´ë‹¤. <code>days_in_milk</code> í˜•ì§ˆì˜ ì „ì²´ ë³€ì´ê°€ ì–´ë–¤ ìš”ì¸ë“¤ë¡œ êµ¬ì„±ë˜ì–´ ìˆëŠ”ì§€ ë³´ì—¬ì¤€ë‹¤.</p>

    <div class="gwas-dl">
      <div class="gwas-kpis">
        <span class="kpi">scanID Variance: <b>3794.3</b> <span class="badge">Vg</span></span>
        <span class="kpi">farm_id Variance: <b>289.4</b> <span class="badge">Vc</span></span>
        <span class="kpi">Residual Variance: <b>17284.7</b> <span class="badge">Ve</span></span>
      </div>

      <dl class="gwas-dl">
        <dt>scanID (ìœ ì „ íš¨ê³¼)</dt>
        <dd>ê°œì²´ ê°„ ìœ ì „ì  ì°¨ì´(í˜ˆì—°ê´€ê³„ ê³ ë ¤)ë¡œ ì¸í•œ ë¶„ì‚° â†’ <b>ìœ ì „ ë¶„ì‚°(Vg)</b>.</dd>
        <dt>farm_id (ë†ì¥ íš¨ê³¼)</dt>
        <dd>ë†ì¥ ê°„ í™˜ê²½ ì°¨ì´ë¡œ ì¸í•œ ë¶„ì‚° â†’ <b>ê³µí†µ í™˜ê²½ ë¶„ì‚°(Vc)</b>.</dd>
        <dt>Residual (ì”ì°¨)</dt>
        <dd>ìœ ì „ ë° ë†ì¥ íš¨ê³¼ë¡œ ì„¤ëª…ë˜ì§€ ì•ŠëŠ” ë‚˜ë¨¸ì§€ ì˜¤ì°¨ â†’ <b>í™˜ê²½ ë¶„ì‚°(Ve)</b>.</dd>
      </dl>
    </div>

    <hr class="hr-soft"/>

    <h4 style="margin:.2rem 0 .8rem;">ğŸ“ 2. ìœ ì „ë ¥ (Heritability) ê³„ì‚°</h4>
    <div class="gwas-grid">
      <div>
        <table class="gwas-table">
          <thead><tr><th>ì„±ë¶„</th><th>ê°’</th><th>ì„¤ëª…</th></tr></thead>
          <tbody>
            <tr><td>Vg</td><td>3794.3</td><td>ìœ ì „ ë¶„ì‚°</td></tr>
            <tr><td>Vc</td><td>289.4</td><td>ë†ì¥(ê³µí†µ í™˜ê²½) ë¶„ì‚°</td></tr>
            <tr><td>Ve</td><td>17284.7</td><td>ì”ì°¨(í™˜ê²½) ë¶„ì‚°</td></tr>
            <tr><td>Vp = Vg + Vc + Ve</td><td><b>21368.4</b></td><td>ì „ì²´ í‘œí˜„í˜• ë¶„ì‚°</td></tr>
            <tr><td>hÂ² = Vg / Vp</td><td><b>0.178</b></td><td>ìœ ì „ë ¥</td></tr>
          </tbody>
        </table>
      </div>
      <div class="gwas-note">
        <p><b>ê²°ë¡ :</b> <code>days_in_milk</code>ì˜ <b>SNP ìœ ì „ë ¥ì€ ì•½ 17.8%</b>ë¡œ ì¶”ì •ëœë‹¤. ì´ëŠ” ì´ í˜•ì§ˆì˜ ì „ì²´ ë³€ì´ ì¤‘ ì•½ 17.8%ê°€ ìš°ë¦¬ê°€ ì¸¡ì •í•œ SNP ë§ˆì»¤ë¡œ ì„¤ëª…ë˜ëŠ” ìœ ì „ì  ìš”ì¸ì— ì˜í•´ ê²°ì •ëœë‹¤ëŠ” ì˜ë¯¸ì´ë‹¤.</p>
        <p>ì¶”ê°€ë¡œ, ë†ì¥ í™˜ê²½ì´ ì„¤ëª…í•˜ëŠ” ë³€ì´ì˜ ë¹„ìœ¨ì€ 289.4 / 21368.4 â‰ˆ <b>1.4%</b>ë¡œ ìƒëŒ€ì ìœ¼ë¡œ ì‘ë‹¤ëŠ” ê²ƒì„ ì•Œ ìˆ˜ ìˆë‹¤.</p>
      </div>
    </div>
  </div>
</div>
<div id="í•¨ìˆ˜ë¥¼-ì´ìš©í•œ-ìœ ì „ë ¥-ê³„ì‚°" class="section level3 hasAnchor" number="3.7.1">
<h3><span class="header-section-number">3.7.1</span> í•¨ìˆ˜ë¥¼ ì´ìš©í•œ ìœ ì „ë ¥ ê³„ì‚°<a href="#%ED%95%A8%EC%88%98%EB%A5%BC-%EC%9D%B4%EC%9A%A9%ED%95%9C-%EC%9C%A0%EC%A0%84%EB%A0%A5-%EA%B3%84%EC%82%B0" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="gwas-ì ˆì°¨.html#cb50-1" tabindex="-1"></a><span class="co"># VarCorr() í•¨ìˆ˜ë¡œ ëª¨ë¸(&#39;mix&#39;)ì—ì„œ ë¶„ì‚° ì„±ë¶„(Variance Components)ì„ ì¶”ì¶œ</span></span>
<span id="cb50-2"><a href="gwas-ì ˆì°¨.html#cb50-2" tabindex="-1"></a>vc <span class="ot">&lt;-</span> <span class="fu">VarCorr</span>(mix)</span>
<span id="cb50-3"><a href="gwas-ì ˆì°¨.html#cb50-3" tabindex="-1"></a><span class="fu">print</span>(vc, <span class="at">comp =</span> <span class="fu">c</span>(<span class="st">&quot;Variance&quot;</span>))</span></code></pre></div>
<pre><code>#&gt;  Groups   Name        Variance
#&gt;  scanID   (Intercept)  3794.3 
#&gt;  farm_id  (Intercept)   289.4 
#&gt;  Residual             17284.7</code></pre>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="gwas-ì ˆì°¨.html#cb52-1" tabindex="-1"></a><span class="co"># ì”ì°¨ ë¶„ì‚°(Residual variance)ì„ ì¶”ì¶œ</span></span>
<span id="cb52-2"><a href="gwas-ì ˆì°¨.html#cb52-2" tabindex="-1"></a><span class="co"># VarCorr ê²°ê³¼ì˜ ì†ì„±(attribute) &quot;sc&quot;ëŠ” ì”ì°¨ì˜ í‘œì¤€í¸ì°¨ì´ë¯€ë¡œ ì œê³±í•˜ì—¬ ë¶„ì‚°ì„ êµ¬í•œë‹¤. </span></span>
<span id="cb52-3"><a href="gwas-ì ˆì°¨.html#cb52-3" tabindex="-1"></a>ve <span class="ot">&lt;-</span> <span class="fu">attr</span>(<span class="fu">VarCorr</span>(mix), <span class="st">&quot;sc&quot;</span>)<span class="sc">^</span><span class="dv">2</span></span>
<span id="cb52-4"><a href="gwas-ì ˆì°¨.html#cb52-4" tabindex="-1"></a></span>
<span id="cb52-5"><a href="gwas-ì ˆì°¨.html#cb52-5" tabindex="-1"></a><span class="co"># SNP ìœ ì „ë ¥(h2) ê³„ì‚°</span></span>
<span id="cb52-6"><a href="gwas-ì ˆì°¨.html#cb52-6" tabindex="-1"></a><span class="co"># h2 = ìœ ì „ë¶„ì‚° / (ìœ ì „ë¶„ì‚° + ë†ì¥ë¶„ì‚° + ì”ì°¨ë¶„ì‚°)</span></span>
<span id="cb52-7"><a href="gwas-ì ˆì°¨.html#cb52-7" tabindex="-1"></a>h2 <span class="ot">&lt;-</span> vc<span class="sc">$</span>scanID <span class="sc">/</span> (vc<span class="sc">$</span>scanID <span class="sc">+</span>  vc<span class="sc">$</span>farm_id <span class="sc">+</span> ve)</span>
<span id="cb52-8"><a href="gwas-ì ˆì°¨.html#cb52-8" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste</span>(<span class="st">&quot;SNP Heritability (h2):&quot;</span>, base<span class="sc">::</span><span class="fu">as.numeric</span>(h2)))</span></code></pre></div>
<pre><code>#&gt; [1] &quot;SNP Heritability (h2): 0.177565708276822&quot;</code></pre>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="gwas-ì ˆì°¨.html#cb54-1" tabindex="-1"></a><span class="co"># ì „ì²´ í‘œí˜„í˜• ë¶„ì‚° ì¤‘ ë†ì¥ íš¨ê³¼(farm ID)ê°€ ì°¨ì§€í•˜ëŠ” ë¶„ì‚° ë¹„ìœ¨ ê³„ì‚°</span></span>
<span id="cb54-2"><a href="gwas-ì ˆì°¨.html#cb54-2" tabindex="-1"></a>farm_var <span class="ot">&lt;-</span> vc<span class="sc">$</span>farm_id <span class="sc">/</span> (vc<span class="sc">$</span>scanID <span class="sc">+</span> vc<span class="sc">$</span>farm_id <span class="sc">+</span> ve)</span>
<span id="cb54-3"><a href="gwas-ì ˆì°¨.html#cb54-3" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste</span>(<span class="st">&quot;Farm Variance ratio:&quot;</span>, base<span class="sc">::</span><span class="fu">as.numeric</span>(farm_var)))</span></code></pre></div>
<pre><code>#&gt; [1] &quot;Farm Variance ratio: 0.0135431802521894&quot;</code></pre>
</div>
</div>
<div id="gwas-ìµœì¢…-null-ëª¨í˜•-ì í•©-ë°-gwas" class="section level2 hasAnchor" number="3.8">
<h2><span class="header-section-number">3.8</span> GWAS ìµœì¢… null ëª¨í˜• ì í•© ë° GWAS<a href="#gwas-%EC%B5%9C%EC%A2%85-null-%EB%AA%A8%ED%98%95-%EC%A0%81%ED%95%A9-%EB%B0%8F-gwas" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="gwas-ì ˆì°¨.html#cb56-1" tabindex="-1"></a><span class="co"># --- ë¶„ì„í•  í˜•ì§ˆì„ ì´ ë³€ìˆ˜ì— ì§€ì • ---</span></span>
<span id="cb56-2"><a href="gwas-ì ˆì°¨.html#cb56-2" tabindex="-1"></a>single_trait <span class="ot">&lt;-</span> <span class="st">&quot;days_in_milk&quot;</span></span>
<span id="cb56-3"><a href="gwas-ì ˆì°¨.html#cb56-3" tabindex="-1"></a></span>
<span id="cb56-4"><a href="gwas-ì ˆì°¨.html#cb56-4" tabindex="-1"></a><span class="co"># ë¶„ì„í•  í˜•ì§ˆì´ ë°ì´í„°ì— ì¡´ì¬í•˜ëŠ”ì§€ í™•ì¸. ì—†ìœ¼ë©´ ì—ëŸ¬ë¥¼ ë°œìƒì‹œí‚¤ê³  ì¤‘ë‹¨</span></span>
<span id="cb56-5"><a href="gwas-ì ˆì°¨.html#cb56-5" tabindex="-1"></a><span class="fu">stopifnot</span>(single_trait <span class="sc">%in%</span> <span class="fu">colnames</span>(scanAnno<span class="sc">@</span>data))</span>
<span id="cb56-6"><a href="gwas-ì ˆì°¨.html#cb56-6" tabindex="-1"></a></span>
<span id="cb56-7"><a href="gwas-ì ˆì°¨.html#cb56-7" tabindex="-1"></a><span class="co"># ê·€ë¬´ëª¨í˜•</span></span>
<span id="cb56-8"><a href="gwas-ì ˆì°¨.html#cb56-8" tabindex="-1"></a>null_single <span class="ot">&lt;-</span> GENESIS<span class="sc">::</span><span class="fu">fitNullModel</span>(</span>
<span id="cb56-9"><a href="gwas-ì ˆì°¨.html#cb56-9" tabindex="-1"></a>  <span class="at">x        =</span> scanAnno,</span>
<span id="cb56-10"><a href="gwas-ì ˆì°¨.html#cb56-10" tabindex="-1"></a>  <span class="at">outcome  =</span> single_trait,</span>
<span id="cb56-11"><a href="gwas-ì ˆì°¨.html#cb56-11" tabindex="-1"></a>  <span class="at">cov.mat  =</span> <span class="fu">list</span>(<span class="at">gen =</span> KI),</span>
<span id="cb56-12"><a href="gwas-ì ˆì°¨.html#cb56-12" tabindex="-1"></a>  <span class="at">family   =</span> <span class="st">&quot;gaussian&quot;</span></span>
<span id="cb56-13"><a href="gwas-ì ˆì°¨.html#cb56-13" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>#&gt; [1] 10779.434372 10779.434372 -1278.687547     1.071434
#&gt; [1]  1144.527824 18338.039393 -1276.652397     1.094402
#&gt; [1]  3645.393100 17248.267336 -1275.789581     1.019586
#&gt; [1]  4223.460337 17110.722529 -1275.775915     1.000707
#&gt; [1]  4101.06852 17235.09102 -1275.77526     1.00002
#&gt; [1]  4140.354608 17200.070348 -1275.775190     1.000002
#&gt; [1]  4128.278 17210.979 -1275.775     1.000
#&gt; [1]  4132.027 17207.607 -1275.775     1.000</code></pre>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="gwas-ì ˆì°¨.html#cb58-1" tabindex="-1"></a><span class="co"># ëª¨í˜•ì—ì„œ ì¶”ì •ëœ ë¶„ì‚° ì„±ë¶„(ìœ ì „ë¶„ì‚°, ì”ì°¨ë¶„ì‚°)ì„ í™•ì¸</span></span>
<span id="cb58-2"><a href="gwas-ì ˆì°¨.html#cb58-2" tabindex="-1"></a>null_single<span class="sc">$</span>varComp </span></code></pre></div>
<pre><code>#&gt;       V_gen V_resid.var 
#&gt;    4132.027   17207.607</code></pre>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="gwas-ì ˆì°¨.html#cb60-1" tabindex="-1"></a><span class="co"># SNP heritability ê³„ì‚° (h2, CI value)</span></span>
<span id="cb60-2"><a href="gwas-ì ˆì°¨.html#cb60-2" tabindex="-1"></a>GENESIS<span class="sc">::</span><span class="fu">varCompCI</span>(null_single, <span class="at">prop =</span> <span class="cn">TRUE</span>)                             </span></code></pre></div>
<pre><code>#&gt;             Proportion    Lower 95  Upper 95
#&gt; V_gen        0.1936316 -0.06226513 0.4495283
#&gt; V_resid.var  0.8063684  0.55047174 1.0622651</code></pre>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="gwas-ì ˆì°¨.html#cb62-1" tabindex="-1"></a><span class="co"># GWAS ì‹¤í–‰</span></span>
<span id="cb62-2"><a href="gwas-ì ˆì°¨.html#cb62-2" tabindex="-1"></a><span class="co"># ìœ ì „ì²´ ë°ì´í„°ë¥¼ **ë¸”ë¡ ë‹¨ìœ„(ì—¬ê¸°ì„œëŠ” 10,000 SNPì”©)**ë¡œ ìˆœì°¨ ì²˜ë¦¬í•˜ê¸° ìœ„í•œ ì´í„°ë ˆì´í„° ìƒì„±: ë©”ëª¨ë¦¬ ì‚¬ìš©ì„ ì¤„ì´ê¸° ìœ„í•¨</span></span>
<span id="cb62-3"><a href="gwas-ì ˆì°¨.html#cb62-3" tabindex="-1"></a>genoIt <span class="ot">&lt;-</span> GWASTools<span class="sc">::</span><span class="fu">GenotypeBlockIterator</span>(genoData, <span class="at">snpBlock =</span> <span class="dv">10000</span>)</span>
<span id="cb62-4"><a href="gwas-ì ˆì°¨.html#cb62-4" tabindex="-1"></a><span class="co"># ìœ„ì—ì„œ ì í•©í•œ ê·€ë¬´ëª¨í˜•ì„ ê³ ì •í•œ ìƒíƒœì—ì„œ, ê° SNPì— ëŒ€í•´ ë‹¨ì¼ë³€ëŸ‰ Score testë¡œ ì—°ê´€ì„±ì„ ê³„ì‚°</span></span>
<span id="cb62-5"><a href="gwas-ì ˆì°¨.html#cb62-5" tabindex="-1"></a>assoc_single <span class="ot">&lt;-</span> GENESIS<span class="sc">::</span><span class="fu">assocTestSingle</span>(</span>
<span id="cb62-6"><a href="gwas-ì ˆì°¨.html#cb62-6" tabindex="-1"></a>  genoIt, <span class="at">null.model =</span> null_single, <span class="at">BPPARAM =</span> BiocParallel<span class="sc">::</span><span class="fu">SerialParam</span>()</span>
<span id="cb62-7"><a href="gwas-ì ˆì°¨.html#cb62-7" tabindex="-1"></a>)</span>
<span id="cb62-8"><a href="gwas-ì ˆì°¨.html#cb62-8" tabindex="-1"></a></span>
<span id="cb62-9"><a href="gwas-ì ˆì°¨.html#cb62-9" tabindex="-1"></a><span class="co"># ---- P-ê°’ ë³´ì • &amp; ìœ ì˜ SNP ì„ ë³„ ---------------------------------</span></span>
<span id="cb62-10"><a href="gwas-ì ˆì°¨.html#cb62-10" tabindex="-1"></a>assoc_single<span class="sc">$</span>p_adj_BH<span class="ot">=</span><span class="fu">p.adjust</span>(assoc_single<span class="sc">$</span>Score.pval, <span class="at">method=</span><span class="st">&quot;BH&quot;</span>)</span>
<span id="cb62-11"><a href="gwas-ì ˆì°¨.html#cb62-11" tabindex="-1"></a>assoc_single<span class="sc">$</span>p_adj_bonferroni<span class="ot">=</span><span class="fu">p.adjust</span>(assoc_single<span class="sc">$</span>Score.pval, <span class="at">method=</span><span class="st">&quot;bonferroni&quot;</span>)</span>
<span id="cb62-12"><a href="gwas-ì ˆì°¨.html#cb62-12" tabindex="-1"></a></span>
<span id="cb62-13"><a href="gwas-ì ˆì°¨.html#cb62-13" tabindex="-1"></a><span class="co"># ì €ì¥</span></span>
<span id="cb62-14"><a href="gwas-ì ˆì°¨.html#cb62-14" tabindex="-1"></a><span class="fu">saveRDS</span>(assoc_single, <span class="at">file =</span> <span class="fu">file.path</span>(gwas_dir, <span class="fu">paste0</span>(<span class="st">&quot;assoc_&quot;</span>, single_trait, <span class="st">&quot;.rds&quot;</span>)))</span>
<span id="cb62-15"><a href="gwas-ì ˆì°¨.html#cb62-15" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_csv</span>(assoc_single, <span class="fu">file.path</span>(tables_dir, <span class="fu">paste0</span>(<span class="st">&quot;assoc_&quot;</span>, single_trait, <span class="st">&quot;.csv&quot;</span>)))</span>
<span id="cb62-16"><a href="gwas-ì ˆì°¨.html#cb62-16" tabindex="-1"></a></span>
<span id="cb62-17"><a href="gwas-ì ˆì°¨.html#cb62-17" tabindex="-1"></a><span class="fu">head</span>(assoc_single, <span class="dv">5</span>)</span></code></pre></div>
<div style="border: 1px solid #ddd; padding: 5px; overflow-x: scroll; width:100%; ">
<table class="table table-striped table-hover table-condensed" style="width: auto !important; margin-left: auto; margin-right: auto;border-bottom: 0;">
<caption>
<span id="tab:assoc-single-table">Table 3.1: </span><span id="tab:assoc-single-table">Table 3.2: </span>GWAS ì—°ê´€ ë¶„ì„ ê²°ê³¼ (5ê°œ SNP ì˜ˆì‹œ)
</caption>
<thead>
<tr>
<th style="text-align:right;">
variant.id <sup>a</sup>
</th>
<th style="text-align:left;">
chr <sup>b</sup>
</th>
<th style="text-align:right;">
pos <sup>c</sup>
</th>
<th style="text-align:right;">
n.obs <sup>d</sup>
</th>
<th style="text-align:right;">
freq <sup>e</sup>
</th>
<th style="text-align:right;">
MAC <sup>f</sup>
</th>
<th style="text-align:right;">
Score <sup>g</sup>
</th>
<th style="text-align:right;">
Score.SE <sup>h</sup>
</th>
<th style="text-align:right;">
Score.Stat <sup>i</sup>
</th>
<th style="text-align:right;">
Score.pval <sup>j</sup>
</th>
<th style="text-align:right;">
Est <sup>k</sup>
</th>
<th style="text-align:right;">
Est.SE <sup>l</sup>
</th>
<th style="text-align:right;">
PVE <sup>m</sup>
</th>
<th style="text-align:right;">
p_adj_BH <sup>n</sup>
</th>
<th style="text-align:right;">
p_adj_bonferroni <sup>o</sup>
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:right;">
135098
</td>
<td style="text-align:right;">
200
</td>
<td style="text-align:right;">
0.615
</td>
<td style="text-align:right;">
154
</td>
<td style="text-align:right;">
-0.124
</td>
<td style="text-align:right;">
0.067
</td>
<td style="text-align:right;">
-1.858
</td>
<td style="text-align:right;">
0.063
</td>
<td style="text-align:right;">
-27.866
</td>
<td style="text-align:right;">
15.002
</td>
<td style="text-align:right;">
0.017
</td>
<td style="text-align:right;">
0.991
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:right;">
267940
</td>
<td style="text-align:right;">
200
</td>
<td style="text-align:right;">
0.065
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:right;">
-0.009
</td>
<td style="text-align:right;">
0.032
</td>
<td style="text-align:right;">
-0.293
</td>
<td style="text-align:right;">
0.770
</td>
<td style="text-align:right;">
-9.212
</td>
<td style="text-align:right;">
31.457
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
1.000
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:right;">
353745
</td>
<td style="text-align:right;">
200
</td>
<td style="text-align:right;">
0.072
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:right;">
-0.035
</td>
<td style="text-align:right;">
0.035
</td>
<td style="text-align:right;">
-1.017
</td>
<td style="text-align:right;">
0.309
</td>
<td style="text-align:right;">
-29.238
</td>
<td style="text-align:right;">
28.762
</td>
<td style="text-align:right;">
0.005
</td>
<td style="text-align:right;">
1.000
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:right;">
4
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:right;">
393248
</td>
<td style="text-align:right;">
200
</td>
<td style="text-align:right;">
0.215
</td>
<td style="text-align:right;">
86
</td>
<td style="text-align:right;">
-0.107
</td>
<td style="text-align:right;">
0.054
</td>
<td style="text-align:right;">
-1.984
</td>
<td style="text-align:right;">
0.047
</td>
<td style="text-align:right;">
-36.830
</td>
<td style="text-align:right;">
18.561
</td>
<td style="text-align:right;">
0.020
</td>
<td style="text-align:right;">
0.991
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:right;">
5
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:right;">
471078
</td>
<td style="text-align:right;">
200
</td>
<td style="text-align:right;">
0.292
</td>
<td style="text-align:right;">
117
</td>
<td style="text-align:right;">
-0.145
</td>
<td style="text-align:right;">
0.061
</td>
<td style="text-align:right;">
-2.396
</td>
<td style="text-align:right;">
0.017
</td>
<td style="text-align:right;">
-39.544
</td>
<td style="text-align:right;">
16.506
</td>
<td style="text-align:right;">
0.029
</td>
<td style="text-align:right;">
0.991
</td>
<td style="text-align:right;">
1
</td>
</tr>
</tbody>
<tfoot>
<tr>
<td style="padding: 0; " colspan="100%">
<sup>a</sup> SNP ê³ ìœ  ID (ì¸ë±ìŠ¤) <sup>b</sup> ì—¼ìƒ‰ì²´ ë²ˆí˜¸ <sup>c</sup> ì—¼ìƒ‰ì²´ ë‚´ ìœ„ì¹˜ (base pair) <sup>d</sup> í•´ë‹¹ SNP ë¶„ì„ì— ì‚¬ìš©ëœ ê°œì²´ ìˆ˜ <sup>e</sup> ëŒ€ë¦½ìœ ì „ì ë¹ˆë„ (Allele Frequency) <sup>f</sup> ë¶€ëŒ€ë¦½ìœ ì „ì ê°œìˆ˜ (Minor Allele Count) <sup>g</sup> ì—°ê´€ì„± ê²€ì •ì˜ Score í†µê³„ëŸ‰ <sup>h</sup> Score í†µê³„ëŸ‰ì˜ í‘œì¤€ ì˜¤ì°¨ <sup>i</sup> í‘œì¤€í™”ëœ Score í†µê³„ëŸ‰ (Z-ê°’) <sup>j</sup> ì—°ê´€ì„± ê²€ì •ì˜ P-ê°’ <sup>k</sup> ëŒ€ë¦½ìœ ì „ìì˜ íš¨ê³¼ ì¶”ì •ì¹˜ (Effect Size) <sup>l</sup> íš¨ê³¼ ì¶”ì •ì¹˜ì˜ í‘œì¤€ ì˜¤ì°¨ <sup>m</sup> í•´ë‹¹ SNPì´ ì„¤ëª…í•˜ëŠ” í‘œí˜„í˜• ë¶„ì‚°ì˜ ë¹„ìœ¨ (PVE) <sup>n</sup> BH ë°©ë²•ìœ¼ë¡œ ë³´ì •ëœ P-ê°’ (FDR) <sup>o</sup> Bonferroni ë°©ë²•ìœ¼ë¡œ ë³´ì •ëœ P-ê°’
</td>
</tr>
</tfoot>
</table>
</div>
<div id="nullmodel-explainer" class="section level3 hasAnchor" number="3.8.1">
<h3><span class="header-section-number">3.8.1</span> Null modelì´ë€? ì™œ ë¨¼ì € ì í•©í•˜ëŠ”ê°€?<a href="gwas-ì ˆì°¨.html#nullmodel-explainer" class="anchor-section" aria-label="Anchor link to header"></a></h3>
</div>
<div class="alert alert-info null-explainer">
<div id="ëª¨í˜•ì‹" class="section level3 hasAnchor" number="3.8.2">
<h3><span class="header-section-number">3.8.2</span> ğŸ“ ëª¨í˜•ì‹<a href="#%EB%AA%A8%ED%98%95%EC%8B%9D" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>ì „ì²´(í˜¼í•©) ëª¨í˜•<br />
<span class="math display">\[
y = X\beta + Zu + G\gamma + \varepsilon
\]</span></p>
<p>Null ëª¨í˜•(ê²€ì •í•  SNP ì œê±°)<br />
<span class="math display">\[
y = X\beta + Zu + \varepsilon
\]</span></p>
<p><strong>ëª¨í˜•í•­ í’€ì´</strong></p>
<ul>
<li><span class="math inline">\(y\)</span> : ê´€ì¸¡í•œ í˜•ì§ˆ ê°’(ì˜ˆ: days_in_milk).<br />
</li>
<li><span class="math inline">\(X\)</span> : <strong>ê³ ì •íš¨ê³¼ ì„¤ê³„í–‰ë ¬(ì…ë ¥ í–‰ë ¬)</strong> â€” ì ˆí¸(1), ì„±ë³„/ì—°ë ¹, ì£¼ì„±ë¶„(PC) ë“±ì˜ ê³µë³€ëŸ‰ ì—´ë¡œ êµ¬ì„±.<br />
</li>
<li><span class="math inline">\(\beta\)</span> : ê³ ì •íš¨ê³¼ ê³„ìˆ˜(ê³µë³€ëŸ‰ì˜ í‰ê· ì ì¸ ì˜í–¥ í¬ê¸°).<br />
</li>
<li><span class="math inline">\(Z\)</span> : <strong>ì„ì˜íš¨ê³¼ ì¸ì‹œë˜ìŠ¤ í–‰ë ¬</strong> â€” ê°œì²´ê°€ ì–´ë–¤ ë ˆë²¨(ê°œì²´ ìì‹ /ë†ì¥ ë“±)ì— ì†í•˜ëŠ”ì§€ 0/1ë¡œ í‘œì‹œ.<br />
</li>
<li><span class="math inline">\(u\)</span> : ì„ì˜íš¨ê³¼ ë²¡í„°.
<ul>
<li>ê°œì²´ ìœ ì „íš¨ê³¼ëŠ” <strong>GRM <span class="math inline">\(K\)</span></strong>(ì¹œì¡±ë„ í–‰ë ¬)ë¡œ ìƒê´€êµ¬ì¡°ë¥¼ ë°˜ì˜í•´ ëª¨ë¸ë§.<br />
</li>
<li>ë†ì¥ íš¨ê³¼ì²˜ëŸ¼ ì„œë¡œ ë…ë¦½Â·ë¶„ì‚° ë™ì¼ë¡œ ë³´ëŠ” íš¨ê³¼ëŠ” <strong>í•­ë“±í–‰ë ¬ <span class="math inline">\(I\)</span></strong>(ìƒê´€ 0, ë¶„ì‚° ë™ì¼)ë¡œ ë‘ .<br />
</li>
</ul></li>
<li><span class="math inline">\(G\)</span> : <strong>ê²€ì • ì¤‘ì¸ ë‹¨ì¼ SNP</strong>ì˜ ìœ ì „ìí˜•(0/1/2 ë˜ëŠ” í†µê³„ì ìœ¼ë¡œ ì¶”ì •ëœ ìœ ì „ìëŸ‰, dosage). <code>assocTestSingle()</code>ê°€ SNPë§ˆë‹¤ ì´ <span class="math inline">\(G\)</span>ë¥¼ ë°”ê¿”ê°€ë©° ì‚¬ìš©.<br />
</li>
<li><span class="math inline">\(\gamma\)</span> : í•´ë‹¹ SNPì˜ íš¨ê³¼ í¬ê¸°(ê·€ë¬´ê°€ì„¤ <span class="math inline">\(H_0:\gamma=0\)</span>).<br />
</li>
<li><span class="math inline">\(\varepsilon\)</span> : ì”ì°¨(ì•„ì§ ì„¤ëª…ë˜ì§€ ì•Šì€ ë³€ë™).</li>
</ul>
</div>
<div id="í•œëˆˆì—-ë³´ê¸°" class="section level3 hasAnchor" number="3.8.3">
<h3><span class="header-section-number">3.8.3</span> ğŸ’¡ í•œëˆˆì— ë³´ê¸°<a href="#%ED%95%9C%EB%88%88%EC%97%90-%EB%B3%B4%EA%B8%B0" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li><strong>Null model</strong>: SNP íš¨ê³¼ ì—†ì´ ê³µë³€ëŸ‰/ì¹œì¡±êµ¬ì¡°ë§Œìœ¼ë¡œ <strong>ì˜¤ì°¨êµ¬ì¡°Â·ë¶„ì‚°ì„±ë¶„</strong>ì„ ë¨¼ì € ì¶”ì •<br />
</li>
<li><strong>GWAS per-SNP</strong>: ìœ„ Nullì„ <strong>ê³ ì •</strong>í•˜ê³  ê° SNPë¥¼ <strong>Score ê³„ì—´ ê²€ì •(Score, Score.SPA ë“±)</strong>ìœ¼ë¡œ ë¹ ë¥´ê²Œ í‰ê°€</li>
</ul>
<p>GWASì—ì„œ Null model(ê·€ë¬´ëª¨í˜•)ì„ ë¨¼ì € ì í•©í•˜ëŠ” ê²ƒì€ <strong>ì •í™•í•˜ê³  íš¨ìœ¨ì ì¸ ë¶„ì„</strong>ì„ ìœ„í•œ í•µì‹¬ ì „ëµì´ë‹¤.</p>
<ol style="list-style-type: decimal">
<li><strong>í†µê³„ì  íƒ€ë‹¹ì„±(ì œ1ì¢… ì˜¤ë¥˜ í†µì œ)</strong>
<ul>
<li>ì§‘ë‹¨ êµ¬ì¡°((population stratification)ë‚˜ ì¹œì¡± ê´€ê³„(cryptic relatedness)ê°€ ìˆìœ¼ë©´, ì´ë¥¼ ë³´ì •í•˜ì§€ ì•Šì„ ë•Œ <strong>genomic inflation(Î»)</strong>ì´ ì»¤ì§€ê³  ê±°ì§“ì–‘ì„±(ì œ1ì¢… ì˜¤ë¥˜)ì´ ëŠ˜ì–´ë‚œë‹¤.<br />
</li>
<li>Null modelì€ <strong>GRM, <span class="math inline">\(K\)</span></strong> ë“±ìœ¼ë¡œ <strong>ì˜¤ì°¨ êµ¬ì¡°ì™€ ë¶„ì‚°ì„±ë¶„</strong>ì„ ë¨¼ì € ì¶”ì •í•´ ì´ êµë€ì„ ì œê±°/ì™„í™”í•œë‹¤. ê·¸ ê²°ê³¼, ì´í›„ SNP ê²€ì •ì—ì„œ <strong>Î»ê°€ ì•ˆì •</strong>ë˜ê³  <strong>ê±°ì§“ì–‘ì„±</strong>ì´ ì–µì œëœë‹¤.</li>
</ul></li>
<li><strong>ê³„ì‚° íš¨ìœ¨ì„±(ì†ë„)</strong>
<ul>
<li>í˜¼í•©ëª¨í˜•ì˜ ë¶„ì‚°ì„±ë¶„ ì¶”ì •ì€ ê³ ë¹„ìš© ì‘ì—…(í° GRM ì—­í–‰ë ¬ ê³„ì‚° ë“±)ì´ë‹¤.<br />
</li>
<li>Nullì—ì„œ <strong>í•œ ë²ˆë§Œ</strong> ë¶„ì‚°ì„±ë¶„ì„ ì¶”ì •Â·ê³ ì •í•´ ë‘ê³ , ì´í›„ ê° SNPëŠ” <strong>Score ê³„ì—´ ê²€ì •</strong>ìœ¼ë¡œ ì²˜ë¦¬í•˜ë©´ <strong>ìˆ˜ì‹­ë§Œ~ìˆ˜ë°±ë§Œ SNP</strong>ë„ ë¹ ë¥´ê²Œ í‰ê°€í•  ìˆ˜ ìˆë‹¤(ë§¤ SNPë§ˆë‹¤ í˜¼í•©ëª¨í˜•ì„ ì¬ì í•©í•  í•„ìš” ì—†ìŒ).</li>
</ul></li>
<li><strong>ê°€ì •(Infinitesimal ê°€ì •)</strong>
<ul>
<li>ì´ ë°©ë²•ì€ â€œ<strong>ê°œë³„ SNP í•˜ë‚˜ê°€ ì „ì²´ ìœ ì „ë¶„ì‚° <span class="math inline">\(\sigma_g^2\)</span>ì— ë¯¸ì¹˜ëŠ” ì˜í–¥ì€ ë§¤ìš° ì‘ë‹¤</strong>â€ëŠ” ê°€ì •ì— ê¸°ë°˜í•œë‹¤.<br />
</li>
<li>ë“œë¬¼ê²Œ íŠ¹ì • SNP íš¨ê³¼ê°€ ë§¤ìš° í¬ë©´(ì˜ˆ: ìœ ì „ë ¥ì˜ í° ë¹„ìœ¨) ì•½ê°„ì˜ ì •í™•ë„ ì†ì‹¤ì´ ìˆì„ ìˆ˜ ìˆìœ¼ë‚˜, ì¼ë°˜ì ìœ¼ë¡œ Nullâ†’Score ì ˆì°¨ê°€ ê°€ì¥ í•©ë¦¬ì ì´ë‹¤.</li>
</ul></li>
</ol>
<div id="genomic-inflation-lambda" class="section level4 hasAnchor" number="3.8.3.1">
<h4><span class="header-section-number">3.8.3.1</span> ğŸ“ˆ Genomic Inflation, <span class="math inline">\(\lambda\)</span><a href="#genomic-inflation-lambda" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li><strong>ì •ì˜</strong>: GWASì—ì„œ ê´€ì¸¡ëœ ê²€ì • í†µê³„ëŸ‰(ë˜ëŠ” P-ê°’)ë“¤ì´ <strong>ê·€ë¬´ê°€ì„¤ í•˜ì˜ ê¸°ëŒ€ì¹˜ë³´ë‹¤ ì „ë°˜ì ìœ¼ë¡œ ë” ìœ ì˜</strong>í•˜ê²Œ ë‚˜ì˜¤ëŠ” í˜„ìƒ. ì´ëŠ” ëª¨ë¸ì´ ë°ì´í„° ì† <strong>ì§‘ë‹¨ ê³„ì¸µí™”</strong>, <strong>ì ì¬ì  ì¹œì¡± ê´€ê³„</strong> ë“±ì„ ì¶©ë¶„íˆ í†µì œí•˜ì§€ ëª»í–ˆì„ ë•Œ ë‚˜íƒ€ë‚˜ëŠ” <strong>ì²´ê³„ì  í¸í–¥(systematic bias)</strong>ì˜ ì‹ í˜¸ì´ë‹¤.</li>
</ul>
<p><strong>ê²°ë¡ </strong>: Null modelì„ ë¨¼ì € ì í•©í•˜ë©´ <strong>genomic inflationì„ ë°©ì§€</strong>í•˜ê³  <strong>ì œ1ì¢… ì˜¤ë¥˜ë¥¼ í†µì œ</strong>í•˜ë©´ì„œ, ëŒ€ê·œëª¨ SNPë¥¼ <strong>ê³ ì†ìœ¼ë¡œ</strong> ë¶„ì„í•  ìˆ˜ ìˆë‹¤.</p>
</div>
</div>
</div>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="gwas-ì ˆì°¨.html#cb63-1" tabindex="-1"></a><span class="co"># ê²°ê³¼ ìš”ì•½</span></span>
<span id="cb63-2"><a href="gwas-ì ˆì°¨.html#cb63-2" tabindex="-1"></a><span class="co"># genomic inflation êµ¬í•˜ê¸°</span></span>
<span id="cb63-3"><a href="gwas-ì ˆì°¨.html#cb63-3" tabindex="-1"></a>chisq  <span class="ot">&lt;-</span> <span class="fu">qchisq</span>(<span class="dv">1</span> <span class="sc">-</span> assoc_single<span class="sc">$</span>Score.pval, <span class="at">df =</span> <span class="dv">1</span>)</span>
<span id="cb63-4"><a href="gwas-ì ˆì°¨.html#cb63-4" tabindex="-1"></a>lambda <span class="ot">&lt;-</span> <span class="fu">median</span>(chisq, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)<span class="sc">/</span>stats<span class="sc">::</span><span class="fu">qchisq</span>(<span class="fl">0.5</span>, <span class="dv">1</span>)</span>
<span id="cb63-5"><a href="gwas-ì ˆì°¨.html#cb63-5" tabindex="-1"></a><span class="co"># ìš”ì•½ í‘œë¡œ ì •ë¦¬í•˜ê¸°</span></span>
<span id="cb63-6"><a href="gwas-ì ˆì°¨.html#cb63-6" tabindex="-1"></a>summary_single <span class="ot">&lt;-</span> tibble<span class="sc">::</span><span class="fu">tibble</span>(</span>
<span id="cb63-7"><a href="gwas-ì ˆì°¨.html#cb63-7" tabindex="-1"></a>  <span class="at">trait =</span> single_trait,</span>
<span id="cb63-8"><a href="gwas-ì ˆì°¨.html#cb63-8" tabindex="-1"></a>  <span class="at">n_snps =</span> <span class="fu">sum</span>(<span class="sc">!</span><span class="fu">is.na</span>(assoc_single<span class="sc">$</span>Score.pval)),</span>
<span id="cb63-9"><a href="gwas-ì ˆì°¨.html#cb63-9" tabindex="-1"></a>  <span class="at">lambda =</span> lambda,</span>
<span id="cb63-10"><a href="gwas-ì ˆì°¨.html#cb63-10" tabindex="-1"></a>  <span class="at">n_sig_BH =</span> <span class="fu">sum</span>(assoc_single<span class="sc">$</span>p_adj_BH <span class="sc">&lt;</span> <span class="fl">0.05</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb63-11"><a href="gwas-ì ˆì°¨.html#cb63-11" tabindex="-1"></a>  <span class="at">n_sig_bonf =</span> <span class="fu">sum</span>(assoc_single<span class="sc">$</span>p_adj_bonferroni <span class="sc">&lt;</span> <span class="fl">0.05</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb63-12"><a href="gwas-ì ˆì°¨.html#cb63-12" tabindex="-1"></a>  <span class="at">min_p =</span> <span class="fu">min</span>(assoc_single<span class="sc">$</span>Score.pval, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb63-13"><a href="gwas-ì ˆì°¨.html#cb63-13" tabindex="-1"></a>  <span class="at">min_p_SNP =</span> assoc_single<span class="sc">$</span>variant.id[<span class="fu">which.min</span>(assoc_single<span class="sc">$</span>Score.pval)],</span>
<span id="cb63-14"><a href="gwas-ì ˆì°¨.html#cb63-14" tabindex="-1"></a>  <span class="at">min_BH =</span> <span class="fu">min</span>(assoc_single<span class="sc">$</span>p_adj_BH, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb63-15"><a href="gwas-ì ˆì°¨.html#cb63-15" tabindex="-1"></a>)</span>
<span id="cb63-16"><a href="gwas-ì ˆì°¨.html#cb63-16" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_csv</span>(summary_single, <span class="fu">file.path</span>(tables_dir, <span class="fu">paste0</span>(<span class="st">&quot;gwas_summary_&quot;</span>, single_trait, <span class="st">&quot;.csv&quot;</span>)))</span></code></pre></div>
<div class="alert alert-success" role="alert">
<h4 class="alert-heading">
ğŸ“ GWAS ìš”ì•½ ê²°ê³¼
</h4>
<table class="table table-striped table-hover table-condensed" style="width: auto !important; margin-left: auto; margin-right: auto;border-bottom: 0;">
<thead>
<tr>
<th style="text-align:left;">
trait <sup>a</sup>
</th>
<th style="text-align:right;">
n_snps <sup>b</sup>
</th>
<th style="text-align:right;">
lambda <sup>c</sup>
</th>
<th style="text-align:right;">
n_sig_BH <sup>d</sup>
</th>
<th style="text-align:right;">
n_sig_bonf <sup>e</sup>
</th>
<th style="text-align:right;">
min_p <sup>f</sup>
</th>
<th style="text-align:right;">
min_p_SNP <sup>g</sup>
</th>
<th style="text-align:right;">
min_BH <sup>h</sup>
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
days_in_milk
</td>
<td style="text-align:right;">
43686
</td>
<td style="text-align:right;">
0.986
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
37174
</td>
<td style="text-align:right;">
0.443
</td>
</tr>
</tbody>
<tfoot>
<tr>
<td style="padding: 0; " colspan="100%">
<sup>a</sup> ë¶„ì„ ëŒ€ìƒ í˜•ì§ˆ ì´ë¦„ <sup>b</sup> ë¶„ì„ì— ì‚¬ìš©ëœ ì´ SNP ê°œìˆ˜ <sup>c</sup> Genomic Inflation Factor (Î»): ì§‘ë‹¨ êµ¬ì¡°í™”ê°€ ì˜ ë³´ì •ë˜ì—ˆëŠ”ì§€ í™•ì¸í•˜ëŠ” ì§€í‘œ. 1ì— ê°€ê¹Œìš¸ìˆ˜ë¡ ì¢‹ìŒ. <sup>d</sup> BH ë³´ì • ê¸°ì¤€(FDR &lt; 0.05)ì„ ë§Œì¡±í•˜ëŠ” ìœ ì˜ SNP ê°œìˆ˜ <sup>e</sup> Bonferroni ë³´ì • ê¸°ì¤€ì„ ë§Œì¡±í•˜ëŠ” ìœ ì˜ SNP ê°œìˆ˜ <sup>f</sup> ê°€ì¥ ë‚®ì€ P-ê°’ (min P-value) <sup>g</sup> ê°€ì¥ ë‚®ì€ P-ê°’ì„ ë³´ì´ëŠ” SNPì˜ index ID <sup>h</sup> ê°€ì¥ ë‚®ì€ BH ë³´ì • P-ê°’ (min FDR)
</td>
</tr>
</tfoot>
</table>
</div>
<style type="text/css">
/* R ì½”ë“œ ì²­í¬(echo=TRUE)ì˜ ìŠ¤íƒ€ì¼ì„ ì§€ì •í•©ë‹ˆë‹¤ */
pre.sourceCode {
  background-color: #e7f3ff !important; /* ì—°í•œ í•˜ëŠ˜ìƒ‰ ë°°ê²½ */
  border: 1px solid #cce0ff !important;   /* ë°°ê²½ìƒ‰ê³¼ ì–´ìš¸ë¦¬ëŠ” ì—°í•œ íŒŒë€ìƒ‰ í…Œë‘ë¦¬ */
  border-radius: 5px;                      /* ëª¨ì„œë¦¬ë¥¼ ì‚´ì§ ë‘¥ê¸€ê²Œ (ì„ íƒ ì‚¬í•­) */
}

/* í…Œì´ë¸” í—¤ë”(th)ì˜ í…ìŠ¤íŠ¸ê°€ ì¤„ë°”ê¿ˆë˜ì§€ ì•Šë„ë¡ ì„¤ì • */
.table th {
  white-space: nowrap;
}
</style>
</div>
<div id="gwas-ê²°ê³¼-ì‹œê°í™”" class="section level2 hasAnchor" number="3.9">
<h2><span class="header-section-number">3.9</span> GWAS ê²°ê³¼ ì‹œê°í™”<a href="#gwas-%EA%B2%B0%EA%B3%BC-%EC%8B%9C%EA%B0%81%ED%99%94" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="gwas-ì ˆì°¨.html#cb64-1" tabindex="-1"></a><span class="co"># PVE ë¶„í¬ í™•ì¸</span></span>
<span id="cb64-2"><a href="gwas-ì ˆì°¨.html#cb64-2" tabindex="-1"></a>p <span class="ot">=</span> <span class="fu">ggplot</span>(assoc_single, <span class="fu">aes</span>(<span class="at">x =</span> PVE<span class="sc">*</span><span class="dv">100</span>)) <span class="sc">+</span> <span class="co"># PVEë¥¼ í¼ì„¼íŠ¸ë¡œ ë³€í™˜</span></span>
<span id="cb64-3"><a href="gwas-ì ˆì°¨.html#cb64-3" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">y =</span> <span class="fu">after_stat</span>(density)), <span class="at">bins =</span> <span class="dv">30</span>, <span class="at">fill =</span> <span class="st">&quot;grey70&quot;</span>, <span class="at">color =</span> <span class="st">&quot;white&quot;</span>) <span class="sc">+</span></span>
<span id="cb64-4"><a href="gwas-ì ˆì°¨.html#cb64-4" tabindex="-1"></a>  <span class="fu">geom_density</span>(<span class="at">linewidth =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb64-5"><a href="gwas-ì ˆì°¨.html#cb64-5" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;Distribution of Variance Explained (PVE) by Single Variants&quot;</span>, <span class="at">x =</span> <span class="st">&quot;PVE (%)&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Density&quot;</span>) <span class="sc">+</span></span>
<span id="cb64-6"><a href="gwas-ì ˆì°¨.html#cb64-6" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb64-7"><a href="gwas-ì ˆì°¨.html#cb64-7" tabindex="-1"></a></span>
<span id="cb64-8"><a href="gwas-ì ˆì°¨.html#cb64-8" tabindex="-1"></a><span class="fu">print</span>(p)</span></code></pre></div>
<p><img src="my-final-book_files/figure-html/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="gwas-ì ˆì°¨.html#cb65-1" tabindex="-1"></a><span class="co"># í”Œë¡¯ ì €ì¥</span></span>
<span id="cb65-2"><a href="gwas-ì ˆì°¨.html#cb65-2" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">file.path</span>(figs_dir, <span class="fu">paste0</span>(<span class="st">&quot;pve_hist_&quot;</span>, single_trait, <span class="st">&quot;.png&quot;</span>)), p, <span class="at">dpi=</span><span class="dv">300</span>, <span class="at">width=</span><span class="dv">6</span>, <span class="at">height=</span><span class="dv">4</span>)</span>
<span id="cb65-3"><a href="gwas-ì ˆì°¨.html#cb65-3" tabindex="-1"></a></span>
<span id="cb65-4"><a href="gwas-ì ˆì°¨.html#cb65-4" tabindex="-1"></a><span class="co"># ---- QQ í”Œë¡¯ &amp; ë§¨í•´íŠ¼ í”Œë¡¯ -------------------------------</span></span>
<span id="cb65-5"><a href="gwas-ì ˆì°¨.html#cb65-5" tabindex="-1"></a>manplot <span class="ot">&lt;-</span> assoc_single <span class="sc">|&gt;</span></span>
<span id="cb65-6"><a href="gwas-ì ˆì°¨.html#cb65-6" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="at">SNP =</span> variant.id, <span class="at">Chromosome =</span> chr, <span class="at">Position =</span> pos, <span class="at">P =</span> Score.pval)</span>
<span id="cb65-7"><a href="gwas-ì ˆì°¨.html#cb65-7" tabindex="-1"></a></span>
<span id="cb65-8"><a href="gwas-ì ˆì°¨.html#cb65-8" tabindex="-1"></a><span class="co"># ì—¼ìƒ‰ì²´ ë¶„ë¥˜ í™•ì¸</span></span>
<span id="cb65-9"><a href="gwas-ì ˆì°¨.html#cb65-9" tabindex="-1"></a>dplyr<span class="sc">::</span><span class="fu">distinct</span>(manplot, Chromosome)</span></code></pre></div>
<pre><code>#&gt;    Chromosome
#&gt; 1           1
#&gt; 2           2
#&gt; 3           3
#&gt; 4           4
#&gt; 5           5
#&gt; 6           6
#&gt; 7           7
#&gt; 8           8
#&gt; 9           9
#&gt; 10         10
#&gt; 11         11
#&gt; 12         12
#&gt; 13         13
#&gt; 14         14
#&gt; 15         15
#&gt; 16         16
#&gt; 17         17
#&gt; 18         18
#&gt; 19         19
#&gt; 20         20
#&gt; 21         21
#&gt; 22         22
#&gt; 23         23
#&gt; 24         24
#&gt; 25         25
#&gt; 26         26
#&gt; 27         27
#&gt; 28         28
#&gt; 29         29
#&gt; 30          U</code></pre>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="gwas-ì ˆì°¨.html#cb67-1" tabindex="-1"></a><span class="co"># U ì—¼ìƒ‰ì²´ ì½”ë”© ë³€ê²½</span></span>
<span id="cb67-2"><a href="gwas-ì ˆì°¨.html#cb67-2" tabindex="-1"></a>manplot <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">mutate</span>(manplot, <span class="at">Chromosome =</span> dplyr<span class="sc">::</span><span class="fu">case_when</span>(Chromosome <span class="sc">==</span> <span class="st">&quot;U&quot;</span> <span class="sc">~</span> <span class="st">&quot;X&quot;</span>, <span class="cn">TRUE</span> <span class="sc">~</span> Chromosome))</span>
<span id="cb67-3"><a href="gwas-ì ˆì°¨.html#cb67-3" tabindex="-1"></a></span>
<span id="cb67-4"><a href="gwas-ì ˆì°¨.html#cb67-4" tabindex="-1"></a><span class="do">## QQ plot</span></span>
<span id="cb67-5"><a href="gwas-ì ˆì°¨.html#cb67-5" tabindex="-1"></a>CMplot<span class="sc">::</span><span class="fu">CMplot</span>(                                                           <span class="co">#</span></span>
<span id="cb67-6"><a href="gwas-ì ˆì°¨.html#cb67-6" tabindex="-1"></a>  <span class="at">Pmap =</span> manplot,</span>
<span id="cb67-7"><a href="gwas-ì ˆì°¨.html#cb67-7" tabindex="-1"></a>  <span class="at">plot.type =</span> <span class="st">&quot;q&quot;</span>,</span>
<span id="cb67-8"><a href="gwas-ì ˆì°¨.html#cb67-8" tabindex="-1"></a>  <span class="at">conf.int =</span> <span class="cn">TRUE</span>,</span>
<span id="cb67-9"><a href="gwas-ì ˆì°¨.html#cb67-9" tabindex="-1"></a>  <span class="at">box =</span> <span class="cn">TRUE</span>,</span>
<span id="cb67-10"><a href="gwas-ì ˆì°¨.html#cb67-10" tabindex="-1"></a>  <span class="at">main =</span> <span class="fu">paste0</span>(<span class="st">&quot;QQ Plot (Î» = &quot;</span>, <span class="fu">round</span>(lambda, <span class="dv">3</span>), <span class="st">&quot;)&quot;</span>),</span>
<span id="cb67-11"><a href="gwas-ì ˆì°¨.html#cb67-11" tabindex="-1"></a>  <span class="at">file =</span> <span class="st">&quot;png&quot;</span>,</span>
<span id="cb67-12"><a href="gwas-ì ˆì°¨.html#cb67-12" tabindex="-1"></a>  <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">width =</span> <span class="dv">4</span>, <span class="at">height =</span> <span class="dv">4</span>,</span>
<span id="cb67-13"><a href="gwas-ì ˆì°¨.html#cb67-13" tabindex="-1"></a>  <span class="co"># â†“â†“â†“ í¬ê¸° ì¡°ì ˆ</span></span>
<span id="cb67-14"><a href="gwas-ì ˆì°¨.html#cb67-14" tabindex="-1"></a>  <span class="at">axis.cex =</span> <span class="fl">0.7</span>,   <span class="co"># ì¶• ëˆˆê¸ˆ ê¸€ì í¬ê¸°(ê¸°ë³¸ 1 â†’ ë” ì‘ê²Œ)</span></span>
<span id="cb67-15"><a href="gwas-ì ˆì°¨.html#cb67-15" tabindex="-1"></a>  <span class="at">lab.cex  =</span> <span class="fl">0.9</span>,   <span class="co"># ì¶• ë¼ë²¨(ì œëª©) í¬ê¸°(ê¸°ë³¸ 1.5 â†’ ë” ì‘ê²Œ)</span></span>
<span id="cb67-16"><a href="gwas-ì ˆì°¨.html#cb67-16" tabindex="-1"></a>  <span class="at">main.cex =</span> <span class="fl">1.0</span>,   <span class="co"># íƒ€ì´í‹€ í¬ê¸°(ê¸°ë³¸ 1.5 â†’ ë” ì‘ê²Œ)</span></span>
<span id="cb67-17"><a href="gwas-ì ˆì°¨.html#cb67-17" tabindex="-1"></a> <span class="co"># ë¼ë²¨ ìœ„ì¹˜ ë¯¸ì„¸ì¡°ì •</span></span>
<span id="cb67-18"><a href="gwas-ì ˆì°¨.html#cb67-18" tabindex="-1"></a>  <span class="at">xticks.pos =</span> <span class="fl">0.5</span>,    <span class="co"># xì¶• ëˆˆê¸ˆê³¼ ì¶• ê°„ê²©</span></span>
<span id="cb67-19"><a href="gwas-ì ˆì°¨.html#cb67-19" tabindex="-1"></a>  <span class="at">ylab.pos   =</span> <span class="fl">1.5</span>,    <span class="co"># yì¶• ë¼ë²¨ê³¼ ì¶• ê°„ê²©</span></span>
<span id="cb67-20"><a href="gwas-ì ˆì°¨.html#cb67-20" tabindex="-1"></a> <span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">3</span>, <span class="dv">4</span>, <span class="fl">2.5</span>, <span class="dv">2</span>)  <span class="co"># ì•„ë˜-ì™¼-ìœ„-ì˜¤ë¥¸ ì—¬ë°±(í´ë¦½ ë°©ì§€/ë°€ì°© ì¡°ì ˆ)</span></span>
<span id="cb67-21"><a href="gwas-ì ˆì°¨.html#cb67-21" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>#&gt;  Q-Q plotting P.</code></pre>
<pre><code>#&gt;  Plots are stored in: G:/Rhome/NIAS_ibv3_296ea/PLINK_120825_0322</code></pre>
<p>
<em>QQ plot</em>
</p>
<p><img src="05_gwas/holstein/figs/QQplot.P.png" width="600" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="gwas-ì ˆì°¨.html#cb70-1" tabindex="-1"></a><span class="do">## Manhattan plot</span></span>
<span id="cb70-2"><a href="gwas-ì ˆì°¨.html#cb70-2" tabindex="-1"></a>CMplot<span class="sc">::</span><span class="fu">CMplot</span>(</span>
<span id="cb70-3"><a href="gwas-ì ˆì°¨.html#cb70-3" tabindex="-1"></a>  manplot,</span>
<span id="cb70-4"><a href="gwas-ì ˆì°¨.html#cb70-4" tabindex="-1"></a>  <span class="at">plot.type =</span> <span class="st">&quot;m&quot;</span>,</span>
<span id="cb70-5"><a href="gwas-ì ˆì°¨.html#cb70-5" tabindex="-1"></a>  <span class="at">LOG10 =</span> <span class="cn">TRUE</span>,</span>
<span id="cb70-6"><a href="gwas-ì ˆì°¨.html#cb70-6" tabindex="-1"></a>  <span class="at">chr.den.col =</span> <span class="cn">NULL</span>,</span>
<span id="cb70-7"><a href="gwas-ì ˆì°¨.html#cb70-7" tabindex="-1"></a>  <span class="at">file =</span> <span class="st">&quot;png&quot;</span>,</span>
<span id="cb70-8"><a href="gwas-ì ˆì°¨.html#cb70-8" tabindex="-1"></a>  <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">width =</span> <span class="dv">12</span>, <span class="at">height =</span> <span class="dv">4</span></span>
<span id="cb70-9"><a href="gwas-ì ˆì°¨.html#cb70-9" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>#&gt;  Rectangular Manhattan plotting P.</code></pre>
<pre><code>#&gt;  Plots are stored in: G:/Rhome/NIAS_ibv3_296ea/PLINK_120825_0322</code></pre>
<p>
<em>Manhattan plot</em>
</p>
<p><img src="05_gwas/holstein/figs/Rect_Manhtn.P.png" width="1800" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="gwas-ì ˆì°¨.html#cb73-1" tabindex="-1"></a><span class="do">## Circular Manhattan plot</span></span>
<span id="cb73-2"><a href="gwas-ì ˆì°¨.html#cb73-2" tabindex="-1"></a>CMplot<span class="sc">::</span><span class="fu">CMplot</span>(</span>
<span id="cb73-3"><a href="gwas-ì ˆì°¨.html#cb73-3" tabindex="-1"></a>  manplot,</span>
<span id="cb73-4"><a href="gwas-ì ˆì°¨.html#cb73-4" tabindex="-1"></a>  <span class="at">plot.type =</span> <span class="st">&quot;c&quot;</span>,</span>
<span id="cb73-5"><a href="gwas-ì ˆì°¨.html#cb73-5" tabindex="-1"></a>  <span class="at">LOG10 =</span> <span class="cn">TRUE</span>,</span>
<span id="cb73-6"><a href="gwas-ì ˆì°¨.html#cb73-6" tabindex="-1"></a>  <span class="at">chr.den.col =</span> <span class="cn">NULL</span>,</span>
<span id="cb73-7"><a href="gwas-ì ˆì°¨.html#cb73-7" tabindex="-1"></a>  <span class="at">file =</span> <span class="st">&quot;png&quot;</span>,</span>
<span id="cb73-8"><a href="gwas-ì ˆì°¨.html#cb73-8" tabindex="-1"></a>  <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">width =</span> <span class="dv">4</span>, <span class="at">height =</span> <span class="dv">4</span>,</span>
<span id="cb73-9"><a href="gwas-ì ˆì°¨.html#cb73-9" tabindex="-1"></a>  <span class="at">axis.cex =</span> <span class="fl">0.7</span>,   <span class="co"># ì¶• ëˆˆê¸ˆ ê¸€ì í¬ê¸°(ê¸°ë³¸ 1 â†’ ë” ì‘ê²Œ)</span></span>
<span id="cb73-10"><a href="gwas-ì ˆì°¨.html#cb73-10" tabindex="-1"></a>  <span class="at">lab.cex  =</span> <span class="fl">0.9</span>,   <span class="co"># ì¶• ë¼ë²¨(ì œëª©) í¬ê¸°(ê¸°ë³¸ 1.5 â†’ ë” ì‘ê²Œ)</span></span>
<span id="cb73-11"><a href="gwas-ì ˆì°¨.html#cb73-11" tabindex="-1"></a>  <span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>),   <span class="co"># ê¸°ë³¸(3,6,3,3)ë³´ë‹¤ í›¨ì”¬ íƒ€ì´íŠ¸</span></span>
<span id="cb73-12"><a href="gwas-ì ˆì°¨.html#cb73-12" tabindex="-1"></a>  <span class="at">cir.chr.h =</span> <span class="fl">0.6</span>              <span class="co"># ë°”ê¹¥ í¬ë¡œëª¨ì†œ í…Œë‘ë¦¬ ë‘ê»˜ ì¶•ì†Œ</span></span>
<span id="cb73-13"><a href="gwas-ì ˆì°¨.html#cb73-13" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>#&gt;  Circular Manhattan plotting P.
#&gt;  Plots are stored in: G:/Rhome/NIAS_ibv3_296ea/PLINK_120825_0322</code></pre>
<p>
<em>Circular Manhattan plot</em>
</p>
<p><img src="05_gwas/holstein/figs/Cir_Manhtn.P.png" width="600" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="gwas-ì ˆì°¨.html#cb75-1" tabindex="-1"></a><span class="co"># plot ìœ„ì¹˜ ë° ì´ë¦„ ë³€ê²½</span></span>
<span id="cb75-2"><a href="gwas-ì ˆì°¨.html#cb75-2" tabindex="-1"></a><span class="fu">file.rename</span>(<span class="st">&quot;QQplot.P.png&quot;</span>, <span class="fu">file.path</span>(figs_dir, <span class="fu">paste0</span>(<span class="st">&quot;QQplot.P_&quot;</span>, single_trait, <span class="st">&quot;.png&quot;</span>)))</span>
<span id="cb75-3"><a href="gwas-ì ˆì°¨.html#cb75-3" tabindex="-1"></a><span class="fu">file.rename</span>(<span class="st">&quot;Rect_Manhtn.P.png&quot;</span>, <span class="fu">file.path</span>(figs_dir, <span class="fu">paste0</span>(<span class="st">&quot;Rect_Manhtn.P_&quot;</span>, single_trait, <span class="st">&quot;.png&quot;</span>)))</span>
<span id="cb75-4"><a href="gwas-ì ˆì°¨.html#cb75-4" tabindex="-1"></a><span class="fu">file.rename</span>(<span class="st">&quot;Cir_Manhtn.P.png&quot;</span>, <span class="fu">file.path</span>(figs_dir, <span class="fu">paste0</span>(<span class="st">&quot;Cir_Manhtn.P_&quot;</span>, single_trait, <span class="st">&quot;.png&quot;</span>)))</span></code></pre></div>
<style>
  /* ì´ ì¶œë ¥ ë¸”ë¡ ì „ì²´ì˜ ê¸°ë³¸ í°íŠ¸/ì¤„ë†’ì´ í†µì¼ */
  .outputs { font-size: .95rem; line-height: 1.5; }

  /* í‘œì™€ í‘œ ë‚´ë¶€ ìš”ì†Œë“¤ì´ ë™ì¼í•œ í¬ê¸°ë¥¼ ìƒì†ë°›ë„ë¡ ê°•ì œ */
  .outputs .outputs-table,
  .outputs .outputs-table td,
  .outputs .outputs-table th,
  .outputs .outputs-table ul,
  .outputs .outputs-table li,
  .outputs .outputs-table code {
    font-size: inherit !important;
    line-height: inherit !important;
  }

  /* í‘œ ì•ˆ ë¦¬ìŠ¤íŠ¸ ì—¬ë°± ì •ë¦¬ */
  .outputs .outputs-table td ul { margin: 0; padding-left: 1rem; }

  /* í‘œ í­ ìë™ */
  .outputs .outputs-table { table-layout: auto !important; width: 100%; }

  /* ì ˆëŒ€ ì¤„ë°”ê¿ˆ ê¸ˆì§€(ì…€ + ë‚´ë¶€ ëª¨ë“  ìš”ì†Œ) */
  .outputs .outputs-table td.nowrap,
  .outputs .outputs-table td.nowrap *,
  .outputs .outputs-table td.nowrap code {
    white-space: nowrap !important;
    word-break: normal !important;
    overflow-wrap: normal !important;
    hyphens: none !important;
  }

  /* Bootstrap ì—†ëŠ” í™˜ê²½ ëŒ€ë¹„ (ì•ˆì „ì¥ì¹˜) */
  .mb-0 { margin-bottom: 0 !important; }
  .list-unstyled { list-style: none; padding-left: 0; }
</style>
<div class="alert alert-success outputs" role="alert">
<h4 class="alert-heading">
ğŸ“ ìµœì¢…ì¶œë ¥ë¬¼
</h4>
<h5 class="mt-2">
í’ˆì¢…ë³„ (<code>05_gwas/&lt;breed&gt;/</code> ë‚´ë¶€)
</h5>
<table class="table table-striped table-condensed outputs-table">
<thead>
<tr>
<th>
ìœ„ì¹˜
</th>
<th>
íŒŒì¼ ì´ë¦„
</th>
<th>
í˜•ì‹
</th>
<th>
ì„¤ëª…
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<code>05_gwas/&lt;breed&gt;/</code>
</td>
<td>
<code>A_matrix_heatmap.png</code>
</td>
<td>
PNG
</td>
<td>
GRM(ìœ ì „ì²´ ê´€ê³„ í–‰ë ¬) íˆíŠ¸ë§µ
</td>
</tr>
<!-- í”Œë¡¯ í–‰: í”Œë¡¯ë³„ 1ì¤„ì”© í‘œì‹œ -->
<tr>
<td>
<code>05_gwas/&lt;breed&gt;/figs/</code>
</td>
<td>
<ul class="list-unstyled mb-0">
<li>
<code>pheno_distribution_&lt;trait&gt;.png</code>
</li>
<li>
<code>pve_hist_&lt;trait&gt;.png</code>
</li>
<li>
<code>QQplot.P_&lt;trait&gt;.png</code>
</li>
<li>
<code>Rect_Manhtn.P_&lt;trait&gt;.png</code>
</li>
<li>
<code>Cir_Manhtn.P_&lt;trait&gt;.png</code>
</li>
</ul>
</td>
<td>
PNG
</td>
<td>
<ul class="list-unstyled mb-0">
<li>
í‘œí˜„í˜• ë¶„í¬ í”Œë¡¯
</li>
<li>
PVE ë¶„í¬ í”Œë¡¯
</li>
<li>
QQ í”Œë¡¯(Î» í¬í•¨)
</li>
<li>
ë§¨í•´íŠ¼ í”Œë¡¯(ì§ì‚¬ê°í˜•)
</li>
<li>
ë§¨í•´íŠ¼ í”Œë¡¯(ì›í˜•)
</li>
</ul>
</td>
</tr>
<!-- GENESIS ì„¤ëª…: í•œ ì¤„ ê³ ì • -->
<tr>
<td>
<code>05_gwas/&lt;breed&gt;/gwas/</code>
</td>
<td>
<code>assoc_&lt;trait&gt;.rds</code>
</td>
<td>
RDS
</td>
<td class="nowrap">
GENESIS::assocTestSingle() ê²°ê³¼Â ê°ì²´Â (ì¬ë¶„ì„/ì¶”ê°€Â ê°€ê³µìš©)
</td>
</tr>
<tr>
<td>
<code>05_gwas/&lt;breed&gt;/tables/</code>
</td>
<td>
<code>assoc_&lt;trait&gt;.csv</code>
</td>
<td>
CSV
</td>
<td>
ë‹¨ì¼ ë§ˆì»¤ ì—°ê´€ì„± í‘œ: P-value, íš¨ê³¼, í‘œì¤€ì˜¤ì°¨, PVE, FDR(BH), Bonferroni ë“±
</td>
</tr>
<tr>
<td>
<code>05_gwas/&lt;breed&gt;/tables/</code>
</td>
<td>
<code>gwas_summary_&lt;trait&gt;.csv</code>
</td>
<td>
CSV
</td>
<td>
ìš”ì•½ ì§€í‘œ: SNP ìˆ˜, Î», ìœ ì˜ SNP ê°œìˆ˜(FDR/Bonferroni), ìµœì†Œ P, í•´ë‹¹ SNP index ID
</td>
</tr>
</tbody>
</table>
<h5 class="mt-3">
ì°¸ê³ 
</h5>
<ul class="list-unstyled mb-0">
<li>
ğŸ”§ í˜„ì¬ ì„¤ì •
<ul class="list-unstyled mb-0" style="margin-left:1.1rem">
<li>
<code>&lt;breed&gt; = holstein</code>
</li>
<li>
<code>&lt;trait&gt; = days_in_milk</code>
</li>
</ul>
</li>
</ul>
</div>
<style>
  /* 'outputs' ìƒì ë‚´ë¶€ì˜ h5ì™€, ê·¸ ë‹¤ìŒì— ì˜¤ëŠ” ul ë° liì˜ í°íŠ¸ë¥¼ ì¡°ì ˆ */
  .outputs h5 {
    font-size: 1.2em !important; /* ì´ ê°’ì„ ì¡°ì ˆí•˜ì—¬ 'ì°¸ê³ ' ê¸€ì”¨ í¬ê¸° ë³€ê²½ */
    font-weight: 700; /* ê¸€ì”¨ë¥¼ ì¢€ ë” êµµê²Œ (ì„ íƒ ì‚¬í•­) */
  }
  
  .outputs h5 + ul,
  .outputs h5 + ul li {
    font-size: 1.2em !important; 
    line-height: inherit !important;
  }
</style>
</div>
<div id="rë¡œ-gwas-ê³¼ì •-ì™„ë£Œ" class="section level2 hasAnchor" number="3.10">
<h2><span class="header-section-number">3.10</span> ğŸ‰ Rë¡œ GWAS ê³¼ì • ì™„ë£Œ<a href="#r%EB%A1%9C-gwas-%EA%B3%BC%EC%A0%95-%EC%99%84%EB%A3%8C" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="alert alert-success" role="alert" style="margin-top:.5rem">
<h4 class="alert-heading" style="margin-top:0;margin-bottom:.5rem">
âœ… Rë¡œ í•˜ëŠ” GWAS ì ˆì°¨ë¥¼ ëê¹Œì§€ ì˜ ë§ˆì³¤ë‹¤!
</h4>
<p style="margin-bottom:.75rem">
ëª¨ë“  ë‹¨ê³„ë¥¼ ê±°ì³ ìµœì¢… ì¶œë ¥ë¬¼ì„ ë§Œë“¤ì—ˆë‹¤. ğŸ€
</p>
<hr style="margin:.75rem 0" />
<p class="mb-0">
ì´ì œ Holsteinì˜ ë‹¤ë¥¸ í˜•ì§ˆë“¤ê³¼ Jersey ë°ì´í„°ì…‹ì— ëŒ€í•´ì„œë„ GWAS ë¶„ì„ì„ í•  ìˆ˜ ìˆë‹¤. ğŸš€ğŸ”¥
</p>
</div>
</div>
<div id="script-full" class="section level2 hasAnchor" number="3.11">
<h2><span class="header-section-number">3.11</span> ìŠ¤í¬ë¦½íŠ¸ ì „ì²´ ë³´ê¸°<a href="gwas-ì ˆì°¨.html#script-full" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><span></span></p>
<div class="alert alert-secondary" role="alert" style="margin-top:1rem">
<p>
<span style="font-size:1.1em">ğŸ“˜ <strong> GWAS R script</strong></span>
</p>
<details>
<summary style="cursor:pointer;font-size:1.15em; font-weight:600">
<code>gwas_workflow.R</code> íŒŒì¼ â€” ğŸ‘† í´ë¦­í•˜ì—¬ ë³´ê¸°
</summary>
<pre><code class="language-r">## 1. íŒ¨í‚¤ì§€ ì„¤ì¹˜ & ë¡œë”©
# ---- ì„¤ì¹˜ì–´ì‹œìŠ¤í„´íŠ¸ -------------------------------------------------------
cran_pkgs <- c("genio", "qqman", "sommer", "lme4breeding", "dplyr", "CMplot", "stringr", "ggplot2", "tibble", "readr")
bioc_pkgs <- c("GENESIS","GWASTools","BiocParallel")

if (!requireNamespace("BiocManager", quietly = TRUE)) install.packages("BiocManager")

install_if_missing <- function(pkgs, installer) {
  need <- pkgs[!vapply(pkgs, requireNamespace, logical(1), quietly = TRUE)]
  if (length(need)) installer(need, ask = FALSE)
}

install_if_missing(cran_pkgs, install.packages)
install_if_missing(bioc_pkgs, BiocManager::install)

# ---- ë¼ì´ë¸ŒëŸ¬ë¦¬ ë¡œë“œ ------------------------------------------------------
library(genio)
library(GENESIS)
library(GWASTools)
library(qqman)
library(CMplot)
library(lme4breeding)
library(sommer)
library(dplyr)
library(stringr)
library(ggplot2)
library(tibble)

## 2. ê²½ë¡œ/ì…ë ¥ ì •ì˜
breed        <- "holstein"
plink_prefix <- "03_qc/NIAS_ibv3_296ea.holstein"
pheno_file   <- "NIAS_ibv3_296ea_pheno.csv"

report_dir <- file.path("05_gwas", breed)
figs_dir   <- file.path(report_dir, "figs")
gwas_dir   <- file.path(report_dir, "gwas")
tables_dir <- file.path(report_dir, "tables")
for (d in c(report_dir, figs_dir, gwas_dir, tables_dir)) dir.create(d, recursive = TRUE, showWarnings = FALSE)

## 3. PLINK binary ë°ì´í„° ë¶ˆëŸ¬ì˜¤ê¸°
# read_plink()ëŠ” SNP x Sample í–‰ë ¬(X), bim(ë§ˆì»¤ì •ë³´), fam(ê°œì²´ì •ë³´)ì„ ë¶ˆëŸ¬ì˜¨ë‹¤.
plink <- read_plink(plink_prefix)

X <- plink$X             # í–‰: SNP, ì—´: ìƒ˜í”Œ, ê°’: 0/1/2
bim <- plink$bim         # chr, id, pos, alt, ref ë“±
fam <- plink$fam         # family/sample id ë“±

# ë¯¸ë¦¬ë³´ê¸°
X[1:5, 1:5]
head(bim)
head(fam)

## 4. í‘œí˜„í˜• ë°ì´í„° ë¡œë”© & ì •ë ¬
pheno <- readr::read_csv(pheno_file, show_col_types = FALSE)

sample_id_chr <- colnames(X)  # PLINK ì—´ì´ ìƒ˜í”Œ ID
scanID <- seq_along(sample_id_chr)

pheno_ord <- pheno[match(sample_id_chr, pheno$individual_id), , drop = FALSE]
stopifnot(!any(is.na(pheno_ord$individual_id)))
pheno_ord$sample_id <- sample_id_chr
pheno_ord$scanID    <- scanID
stopifnot(identical(as.character(pheno_ord$individual_id), pheno_ord$sample_id))
# ë¯¸ë¦¬ë³´ê¸°
str(pheno_ord)

# í‘œí˜„í˜• ë¶„í¬ í™•ì¸
# --- ë¶„ì„í•  í˜•ì§ˆì„ ì´ ë³€ìˆ˜ì— ì§€ì • ---
single_trait <- "days_in_milk"

p=ggplot(pheno_ord, aes(x = .data[[single_trait]])) +
  geom_histogram(aes(y = after_stat(density)), bins = 30, fill = "grey70", color = "white") +
  geom_density(linewidth = 1) +
  labs(
    title = paste("Distribution of", single_trait),
    x = single_trait, 
    y = "Density"
  ) +
  theme_minimal()

# ë³€ìˆ˜ì— ì €ì¥ëœ í”Œë¡¯ í™•ì¸
print(p)
# ggsave() í•¨ìˆ˜ë¥¼ ì‚¬ìš©í•˜ì—¬ í”Œë¡¯ì„ png íŒŒì¼ë¡œ ì €ì¥
ggsave(
  filename = file.path(figs_dir, paste0("pheno_distribution_", single_trait, ".png")),
  plot = p,
  width = 6,
  height = 4,
  dpi = 300
)

## 5. GRM(ìœ ì „ì²´ ê´€ê³„ í–‰ë ¬) ê³„ì‚° & ê°„ë‹¨ ì‹œê°í™”
# A.matì€ -1/0/1 ì½”ë”©ëœ genotype matrix ì´ìš©
G_mat <- t(X)              # ìƒ˜í”Œ x SNP
colnames(G_mat) <- rownames(X)
rownames(G_mat) <- sample_id_chr

G_code <- G_mat - 1        # 0/1/2 -> -1/0/1
KI     <- sommer::A.mat(G_code)
# ìˆ˜ì¹˜ì•ˆì •ì„±(ì•„ì£¼ ì‘ì€ ridge)
diag(KI) <- diag(KI) + 1e-4

# scanID ê¸°ì¤€ì˜ dimnames ì§€ì •
rownames(KI) <- as.character(scanID)
colnames(KI) <- as.character(scanID)

stopifnot(
  identical(rownames(KI), as.character(scanID)),
  identical(colnames(KI), as.character(scanID))
)

# ìƒìœ„ 5x5 í™•ì¸
KI[1:5, 1:5]

# íˆíŠ¸ë§µ
colfunc <- grDevices::colorRampPalette(c("steelblue4","springgreen","yellow"))

png(file.path(report_dir, "A_matrix_heatmap.png"), width=2000, height=1400, res=200)
stats::heatmap(KI, col = colfunc(100), Colv = "Rowv", symmetric = TRUE)
dev.off()

## 6. GWAS ë¶„ì„ìš© ë°ì´í„° ê°ì²´ ë§Œë“¤ê¸°
bim <- cbind(index = seq_len(nrow(bim)), bim)
stopifnot(all.equal(rownames(X), bim$id))
stopifnot(all.equal(colnames(X), pheno_ord$sample_id))

rownames(X) <- bim$index
colnames(X) <- scanID

# GENESIS íŒ¨í‚¤ì§€ì—ì„œ ìš”êµ¬í•˜ëŠ” ë°ì´í„° ê°ì²´ ë§Œë“¤ê¸°
mg <- GWASTools::MatrixGenotypeReader(
  genotype   = X,
  snpID      = bim$index,
  chromosome = as.integer(bim$chr),
  position   = as.integer(bim$pos),
  scanID     = scanID,
  autosomeCode = 1L:29L,
  XchromCode   = 30L, YchromCode = 31L, XYchromCode = 32L, MchromCode = 33L
)
# ìœ ì „ìí˜• ë°ì´í„° ê°ì²´ ìƒì„±
genoData <- GWASTools::GenotypeData(mg)
# í‘œí˜„í˜• ë°ì´í„° ê°ì²´ ìƒì„±
scanAnno <- GWASTools::ScanAnnotationDataFrame(pheno_ord)
# ê°ì²´ êµ¬ì¡° í™•ì¸
str(genoData)
str(scanAnno@data)   

## 7. GWAS ê·€ë¬´ëª¨í˜• ì í•© ë° ìœ ì „ë ¥(SNP heritability) ê³„ì‚°
# --- ë¶„ì„í•  í˜•ì§ˆì„ ì´ ë³€ìˆ˜ì— ì§€ì • ---
single_trait <- "days_in_milk"

# ë¶„ì„í•  í˜•ì§ˆì´ ë°ì´í„°ì— ì¡´ì¬í•˜ëŠ”ì§€ í™•ì¸. ì—†ìœ¼ë©´ ì—ëŸ¬ë¥¼ ë°œìƒì‹œí‚¤ê³  ì¤‘ë‹¨
stopifnot(single_trait %in% colnames(scanAnno@data))

# --- ë°ì´í„° ì „ì²˜ë¦¬ ---

# 1. ëª¨ë¸ì—ì„œ íš¨ê³¼ë¥¼ ë¶„ì„í•  ë³€ìˆ˜ë“¤ì„ factor(ë²”ì£¼í˜• ë³€ìˆ˜)ë¡œ ë³€í™˜
pheno_ord$farm_id <- as.factor(pheno_ord$farm_id)
pheno_ord$scanID <- as.factor(pheno_ord$scanID)

# 2. ë¶„ì„í•  í˜•ì§ˆ(days_in_milk)ì— ê²°ì¸¡ì¹˜(NA)ê°€ ìˆëŠ” ìƒ˜í”Œì„ ì œê±°í•˜ì—¬
#    ìµœì¢… ë¶„ì„ ë°ì´í„°ì…‹ 'PH'ë¥¼ ìƒì„±
#    droplevels() í•¨ìˆ˜ëŠ” PH ë°ì´í„°ì…‹ì— ë” ì´ìƒ ì¡´ì¬í•˜ì§€ ì•ŠëŠ” ìƒ˜í”Œì´ë‚˜ ë†ì¥ì˜ ë ˆë²¨ ì •ë³´ë¥¼ ì œê±°
PH <- droplevels(subset(pheno_ord, !is.na(days_in_milk)))

# 3. Kinship í–‰ë ¬(KI)ì„ ìµœì¢… ë¶„ì„ ë°ì´í„°ì…‹ 'PH'ì— ë§ì¶° ìë¥¸ë‹¤.
#    ë°˜ë“œì‹œ 'PH'ì˜ ìƒ˜í”Œ ëª©ë¡ê³¼ ìˆœì„œë¥¼ ê¸°ì¤€ìœ¼ë¡œ ì˜ë¼ì•¼,
#    í‘œí˜„í˜• ë°ì´í„°ì™€ Kinship ë°ì´í„°ì˜ ìƒ˜í”Œì´ 1:1ë¡œ ì •í™•í•˜ê²Œ ì¼ì¹˜í•œë‹¤.
K_scan <- KI[levels(PH$scanID), levels(PH$scanID)]

# ë†ì¥ íš¨ê³¼ì™€ ê°œì²´ë³„ ìœ ì „ íš¨ê³¼ë¥¼ ì„ì˜ íš¨ê³¼(random effect)ë¡œ ì„¤ì •í•˜ì—¬ ëª¨ë¸ì„ ë§Œë“ ë‹¤.
# lmebreed í•¨ìˆ˜ëŠ” í˜ˆì—°ê´€ê³„(kinship)ë¥¼ ê³ ë ¤í•œ í˜¼í•© ëª¨ë¸ì„ ì í•©í•´ì¤€ë‹¤.
mix <- lme4breeding::lmebreed(
  # ëª¨ë¸ ê³µì‹: days_in_milkë¥¼ ê°œì²´ íš¨ê³¼(scanID)ì™€ ë†ì¥ íš¨ê³¼(farm_id)ë¡œ ì„¤ëª…
  days_in_milk ~ (1|scanID) + (1|farm_id),
  
  # relmat ì¸ì: scanIDì˜ ì„ì˜ íš¨ê³¼ëŠ” K_scan í–‰ë ¬ì— ì •ì˜ëœ ìœ ì „ì  ê´€ê³„ë¥¼ ë”°ë¥´ë„ë¡ ì§€ì •í•œë‹¤.
  relmat  = list(scanID = K_scan),
  
  # verbose=TRUE: ëª¨ë¸ ê³„ì‚° ê³¼ì •ì„ í™”ë©´ì— ì¶œë ¥
  verbose = TRUE,
  
  # data ì¸ì: ëª¨ë¸ì— ì‚¬ìš©í•  ë°ì´í„°ì…‹ìœ¼ë¡œ PHë¥¼ ì§€ì •
  data    = PH
)

# ëª¨ë¸ ë¶„ì„ ê²°ê³¼ ìš”ì•½ë³¸ ì¶œë ¥
summary(mix)

### í•¨ìˆ˜ë¥¼ ì´ìš©í•œ ìœ ì „ë ¥ ê³„ì‚°
# VarCorr() í•¨ìˆ˜ë¡œ ëª¨ë¸('mix')ì—ì„œ ë¶„ì‚° ì„±ë¶„(Variance Components)ì„ ì¶”ì¶œ
vc <- VarCorr(mix)
print(vc, comp = c("Variance"))

# ì”ì°¨ ë¶„ì‚°(Residual variance)ì„ ì¶”ì¶œ
# VarCorr ê²°ê³¼ì˜ ì†ì„±(attribute) "sc"ëŠ” ì”ì°¨ì˜ í‘œì¤€í¸ì°¨ì´ë¯€ë¡œ ì œê³±í•˜ì—¬ ë¶„ì‚°ì„ êµ¬í•œë‹¤. 
ve <- attr(VarCorr(mix), "sc")^2

# SNP ìœ ì „ë ¥(h2) ê³„ì‚°
# h2 = ìœ ì „ë¶„ì‚° / (ìœ ì „ë¶„ì‚° + ë†ì¥ë¶„ì‚° + ì”ì°¨ë¶„ì‚°)
h2 <- vc$scanID / (vc$scanID +  vc$farm_id + ve)
print(paste("SNP Heritability (h2):", base::as.numeric(h2)))

# ì „ì²´ í‘œí˜„í˜• ë¶„ì‚° ì¤‘ ë†ì¥ íš¨ê³¼(farm ID)ê°€ ì°¨ì§€í•˜ëŠ” ë¶„ì‚° ë¹„ìœ¨ ê³„ì‚°
farm_var <- vc$farm_id / (vc$scanID + vc$farm_id + ve)
print(paste("Farm Variance ratio:", base::as.numeric(farm_var)))

## 8. GWAS ìµœì¢… null ëª¨í˜• ì í•© ë° GWAS
# --- ë¶„ì„í•  í˜•ì§ˆì„ ì´ ë³€ìˆ˜ì— ì§€ì • ---
single_trait <- "days_in_milk"

# ë¶„ì„í•  í˜•ì§ˆì´ ë°ì´í„°ì— ì¡´ì¬í•˜ëŠ”ì§€ í™•ì¸. ì—†ìœ¼ë©´ ì—ëŸ¬ë¥¼ ë°œìƒì‹œí‚¤ê³  ì¤‘ë‹¨
stopifnot(single_trait %in% colnames(scanAnno@data))

# ê·€ë¬´ëª¨í˜•
null_single <- GENESIS::fitNullModel(
  x        = scanAnno,
  outcome  = single_trait,
  cov.mat  = list(gen = KI),
  family   = "gaussian"
)

# ëª¨í˜•ì—ì„œ ì¶”ì •ëœ ë¶„ì‚° ì„±ë¶„(ìœ ì „ë¶„ì‚°, ì”ì°¨ë¶„ì‚°)ì„ í™•ì¸
null_single$varComp 
# SNP heritability ê³„ì‚° (h2, CI value)
GENESIS::varCompCI(null_single, prop = TRUE)       

# GWAS ì‹¤í–‰
# ìœ ì „ì²´ ë°ì´í„°ë¥¼ **ë¸”ë¡ ë‹¨ìœ„(ì—¬ê¸°ì„œëŠ” 10,000 SNPì”©)**ë¡œ ìˆœì°¨ ì²˜ë¦¬í•˜ê¸° ìœ„í•œ ì´í„°ë ˆì´í„° ìƒì„±: ë©”ëª¨ë¦¬ ì‚¬ìš©ì„ ì¤„ì´ê¸° ìœ„í•¨
genoIt <- GWASTools::GenotypeBlockIterator(genoData, snpBlock = 10000)
# ìœ„ì—ì„œ ì í•©í•œ ê·€ë¬´ëª¨í˜•ì„ ê³ ì •í•œ ìƒíƒœì—ì„œ, ê° SNPì— ëŒ€í•´ ë‹¨ì¼ë³€ëŸ‰ Score testë¡œ ì—°ê´€ì„±ì„ ê³„ì‚°
assoc_single <- GENESIS::assocTestSingle(
  genoIt, null.model = null_single, BPPARAM = BiocParallel::SerialParam()
)

# ---- P-ê°’ ë³´ì • & ìœ ì˜ SNP ì„ ë³„ ---------------------------------
assoc_single$p_adj_BH=p.adjust(assoc_single$Score.pval, method="BH")
assoc_single$p_adj_bonferroni=p.adjust(assoc_single$Score.pval, method="bonferroni")

# ì €ì¥
saveRDS(assoc_single, file = file.path(gwas_dir, paste0("assoc_", single_trait, ".rds")))
readr::write_csv(assoc_single, file.path(tables_dir, paste0("assoc_", single_trait, ".csv")))

head(assoc_single, 5)

# ê²°ê³¼ ìš”ì•½
# genomic inflation êµ¬í•˜ê¸°
chisq  <- stats::qchisq(1 - assoc_single$Score.pval, df = 1)
lambda <- median(chisq, na.rm = TRUE)/stats::qchisq(0.5, 1)
# ìš”ì•½ í‘œë¡œ ì •ë¦¬í•˜ê¸°
summary_single <- tibble::tibble(
  trait = single_trait,
  n_snps = sum(!is.na(assoc_single$Score.pval)),
  lambda = lambda,
  n_sig_BH = sum(assoc_single$p_adj_BH < 0.05, na.rm = TRUE),
  n_sig_bonf = sum(assoc_single$p_adj_bonferroni < 0.05, na.rm = TRUE),
  min_p = min(assoc_single$Score.pval, na.rm = TRUE),
  min_p_SNP = assoc_single$variant.id[which.min(assoc_single$Score.pval)],
  min_BH = min(assoc_single$p_adj_BH, na.rm = TRUE)
)
readr::write_csv(summary_single, file.path(tables_dir, paste0("gwas_summary_", single_trait, ".csv")))

## 9. GWAS ê²°ê³¼ ì‹œê°í™”
# PVE ë¶„í¬ í™•ì¸
p=ggplot(assoc_single, aes(x = PVE*100)) + # PVEë¥¼ í¼ì„¼íŠ¸ë¡œ ë³€í™˜
  geom_histogram(aes(y = after_stat(density)), bins = 30, fill = "grey70", color = "white") +
  geom_density(linewidth = 1) +
  labs(title = "Distribution of Variance Explained (PVE) by Single Variants", x = "PVE (%)", y = "Density") +
  theme_minimal()

# í”Œë¡¯ ì €ì¥
ggsave(file.path(figs_dir, paste0("pve_hist_", single_trait, ".png")), p, dpi=300, width=6, height=4)

# ---- QQ í”Œë¡¯ & ë§¨í•´íŠ¼ í”Œë¡¯ -------------------------------
manplot <- assoc_single |>
  dplyr::select(SNP = variant.id, Chromosome = chr, Position = pos, P = Score.pval)

# ì—¼ìƒ‰ì²´ ë¶„ë¥˜ í™•ì¸
dplyr::distinct(manplot, Chromosome)

# U ì—¼ìƒ‰ì²´ ì½”ë”© ë³€ê²½
manplot <- dplyr::mutate(manplot, Chromosome = dplyr::case_when(Chromosome == "U" ~ "X", TRUE ~ Chromosome))

## QQ plot
CMplot::CMplot(                                                           #
  Pmap = manplot,
  plot.type = "q",
  conf.int = TRUE,
  box = TRUE,
  main = paste0("QQ Plot (Î» = ", round(lambda, 3), ")"),
  file = "png",
  dpi = 300, width = 4, height = 4,
  # â†“â†“â†“ í¬ê¸° ì¡°ì ˆ
  axis.cex = 0.7,   # ì¶• ëˆˆê¸ˆ ê¸€ì í¬ê¸°(ê¸°ë³¸ 1 â†’ ë” ì‘ê²Œ)
  lab.cex  = 0.9,   # ì¶• ë¼ë²¨(ì œëª©) í¬ê¸°(ê¸°ë³¸ 1.5 â†’ ë” ì‘ê²Œ)
  main.cex = 1.0,   # íƒ€ì´í‹€ í¬ê¸°(ê¸°ë³¸ 1.5 â†’ ë” ì‘ê²Œ)
  # ë¼ë²¨ ìœ„ì¹˜ ë¯¸ì„¸ì¡°ì •
  xticks.pos = 0.5,    # xì¶• ëˆˆê¸ˆê³¼ ì¶• ê°„ê²©
  ylab.pos   = 1.5,    # yì¶• ë¼ë²¨ê³¼ ì¶• ê°„ê²©
  mar = c(3, 4, 2.5, 2)  # ì•„ë˜-ì™¼-ìœ„-ì˜¤ë¥¸ ì—¬ë°±(í´ë¦½ ë°©ì§€/ë°€ì°© ì¡°ì ˆ)
)

## Manhattan plot
CMplot::CMplot(
  manplot,
  plot.type = "m",
  LOG10 = TRUE,
  chr.den.col = NULL,
  file = "png",
  dpi = 300, width = 12, height = 4
)

## Circular Manhattan plot
CMplot::CMplot(
  manplot,
  plot.type = "c",
  LOG10 = TRUE,
  chr.den.col = NULL,
  file = "png",
  dpi = 300, width = 4, height = 4,
  axis.cex = 0.7,   # ì¶• ëˆˆê¸ˆ ê¸€ì í¬ê¸°(ê¸°ë³¸ 1 â†’ ë” ì‘ê²Œ)
  lab.cex  = 0.9,   # ì¶• ë¼ë²¨(ì œëª©) í¬ê¸°(ê¸°ë³¸ 1.5 â†’ ë” ì‘ê²Œ)
  mar = c(1, 1, 1, 1),   # ê¸°ë³¸(3,6,3,3)ë³´ë‹¤ í›¨ì”¬ íƒ€ì´íŠ¸
  cir.chr.h = 0.6              # ë°”ê¹¥ í¬ë¡œëª¨ì†œ í…Œë‘ë¦¬ ë‘ê»˜ ì¶•ì†Œ
)

# plot ìœ„ì¹˜ ë° ì´ë¦„ ë³€ê²½
file.rename("QQplot.P.png", file.path(figs_dir, paste0("QQplot.P_", single_trait, ".png")))
file.rename("Rect_Manhtn.P.png", file.path(figs_dir, paste0("Rect_Manhtn.P_", single_trait, ".png")))
file.rename("Cir_Manhtn.P.png", file.path(figs_dir, paste0("Cir_Manhtn.P_", single_trait, ".png")))
    </code></pre>
</details>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="pre-gwas-ì ˆì°¨-ìœ ì „ìí˜•-ë°ì´í„°-qc.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": null,
    "text": null
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": null,
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
